<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>CaCORE::CaBIO</title>
<link rev="made" href="mailto:" />
</head>

<body style="background-color: white">

<p><a name="__index__"></a></p>
<!-- INDEX BEGIN -->

<ul>

	<li><a href="#taxon">Taxon</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_taxon">Attributes of Taxon</a></li>
		<li><a href="#associations_of_taxon">Associations of Taxon</a></li>
	</ul>

	<li><a href="#agent">Agent</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_agent">Attributes of Agent</a></li>
		<li><a href="#associations_of_agent">Associations of Agent</a></li>
	</ul>

	<li><a href="#chromosome">Chromosome</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_chromosome">Attributes of Chromosome</a></li>
		<li><a href="#associations_of_chromosome">Associations of Chromosome</a></li>
	</ul>

	<li><a href="#gene">Gene</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_gene">Attributes of Gene</a></li>
		<li><a href="#associations_of_gene">Associations of Gene</a></li>
	</ul>

	<li><a href="#vocabulary">Vocabulary</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_vocabulary">Attributes of Vocabulary</a></li>
		<li><a href="#associations_of_vocabulary">Associations of Vocabulary</a></li>
	</ul>

	<li><a href="#protocol">Protocol</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_protocol">Attributes of Protocol</a></li>
		<li><a href="#associations_of_protocol">Associations of Protocol</a></li>
	</ul>

	<li><a href="#tissue">Tissue</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_tissue">Attributes of Tissue</a></li>
		<li><a href="#associations_of_tissue">Associations of Tissue</a></li>
	</ul>

	<li><a href="#snp">SNP</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_snp">Attributes of SNP</a></li>
		<li><a href="#associations_of_snp">Associations of SNP</a></li>
	</ul>

	<li><a href="#genealias">GeneAlias</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_genealias">Attributes of GeneAlias</a></li>
		<li><a href="#associations_of_genealias">Associations of GeneAlias</a></li>
	</ul>

	<li><a href="#library">Library</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_library">Attributes of Library</a></li>
		<li><a href="#associations_of_library">Associations of Library</a></li>
	</ul>

	<li><a href="#clone">Clone</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_clone">Attributes of Clone</a></li>
		<li><a href="#associations_of_clone">Associations of Clone</a></li>
	</ul>

	<li><a href="#target">Target</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_target">Attributes of Target</a></li>
		<li><a href="#associations_of_target">Associations of Target</a></li>
	</ul>

	<li><a href="#location">Location</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_location">Attributes of Location</a></li>
		<li><a href="#associations_of_location">Associations of Location</a></li>
	</ul>

	<li><a href="#clonerelativelocation">CloneRelativeLocation</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_clonerelativelocation">Attributes of CloneRelativeLocation</a></li>
		<li><a href="#associations_of_clonerelativelocation">Associations of CloneRelativeLocation</a></li>
	</ul>

	<li><a href="#nucleicacidsequence">NucleicAcidSequence</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_nucleicacidsequence">Attributes of NucleicAcidSequence</a></li>
		<li><a href="#associations_of_nucleicacidsequence">Associations of NucleicAcidSequence</a></li>
	</ul>

	<li><a href="#diseaseontology">DiseaseOntology</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_diseaseontology">Attributes of DiseaseOntology</a></li>
		<li><a href="#associations_of_diseaseontology">Associations of DiseaseOntology</a></li>
	</ul>

	<li><a href="#clinicaltrialprotocol">ClinicalTrialProtocol</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_clinicaltrialprotocol">Attributes of ClinicalTrialProtocol</a></li>
		<li><a href="#associations_of_clinicaltrialprotocol">Associations of ClinicalTrialProtocol</a></li>
	</ul>

	<li><a href="#protocolassociation">ProtocolAssociation</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_protocolassociation">Attributes of ProtocolAssociation</a></li>
		<li><a href="#associations_of_protocolassociation">Associations of ProtocolAssociation</a></li>
	</ul>

	<li><a href="#pathway">Pathway</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_pathway">Attributes of Pathway</a></li>
		<li><a href="#associations_of_pathway">Associations of Pathway</a></li>
	</ul>

	<li><a href="#organontology">OrganOntology</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_organontology">Attributes of OrganOntology</a></li>
		<li><a href="#associations_of_organontology">Associations of OrganOntology</a></li>
	</ul>

	<li><a href="#organontologyrelationship">OrganOntologyRelationship</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_organontologyrelationship">Attributes of OrganOntologyRelationship</a></li>
		<li><a href="#associations_of_organontologyrelationship">Associations of OrganOntologyRelationship</a></li>
	</ul>

	<li><a href="#histopathology">Histopathology</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_histopathology">Attributes of Histopathology</a></li>
		<li><a href="#associations_of_histopathology">Associations of Histopathology</a></li>
	</ul>

	<li><a href="#geneontology">GeneOntology</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_geneontology">Attributes of GeneOntology</a></li>
		<li><a href="#associations_of_geneontology">Associations of GeneOntology</a></li>
	</ul>

	<li><a href="#geneontologyrelationship">GeneOntologyRelationship</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_geneontologyrelationship">Attributes of GeneOntologyRelationship</a></li>
		<li><a href="#associations_of_geneontologyrelationship">Associations of GeneOntologyRelationship</a></li>
	</ul>

	<li><a href="#diseaseontologyrelationship">DiseaseOntologyRelationship</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_diseaseontologyrelationship">Attributes of DiseaseOntologyRelationship</a></li>
		<li><a href="#associations_of_diseaseontologyrelationship">Associations of DiseaseOntologyRelationship</a></li>
	</ul>

	<li><a href="#anomaly">Anomaly</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_anomaly">Attributes of Anomaly</a></li>
		<li><a href="#associations_of_anomaly">Associations of Anomaly</a></li>
	</ul>

	<li><a href="#populationfrequency">PopulationFrequency</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_populationfrequency">Attributes of PopulationFrequency</a></li>
		<li><a href="#associations_of_populationfrequency">Associations of PopulationFrequency</a></li>
	</ul>

	<li><a href="#proteinsequence">ProteinSequence</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_proteinsequence">Attributes of ProteinSequence</a></li>
		<li><a href="#associations_of_proteinsequence">Associations of ProteinSequence</a></li>
	</ul>

	<li><a href="#protein">Protein</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_protein">Attributes of Protein</a></li>
		<li><a href="#associations_of_protein">Associations of Protein</a></li>
	</ul>

	<li><a href="#proteinalias">ProteinAlias</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_proteinalias">Attributes of ProteinAlias</a></li>
		<li><a href="#associations_of_proteinalias">Associations of ProteinAlias</a></li>
	</ul>

	<li><a href="#homologousassociation">HomologousAssociation</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_homologousassociation">Attributes of HomologousAssociation</a></li>
		<li><a href="#associations_of_homologousassociation">Associations of HomologousAssociation</a></li>
	</ul>

	<li><a href="#physicallocation">PhysicalLocation</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_physicallocation">Attributes of PhysicalLocation</a></li>
		<li><a href="#associations_of_physicallocation">Associations of PhysicalLocation</a></li>
	</ul>

	<li><a href="#cytoband">Cytoband</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_cytoband">Attributes of Cytoband</a></li>
		<li><a href="#associations_of_cytoband">Associations of Cytoband</a></li>
	</ul>

	<li><a href="#cytogeneticlocation">CytogeneticLocation</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_cytogeneticlocation">Attributes of CytogeneticLocation</a></li>
		<li><a href="#associations_of_cytogeneticlocation">Associations of CytogeneticLocation</a></li>
	</ul>

	<li><a href="#generelativelocation">GeneRelativeLocation</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_generelativelocation">Attributes of GeneRelativeLocation</a></li>
		<li><a href="#associations_of_generelativelocation">Associations of GeneRelativeLocation</a></li>
	</ul>

	<li><a href="#genericreporter">GenericReporter</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_genericreporter">Attributes of GenericReporter</a></li>
		<li><a href="#associations_of_genericreporter">Associations of GenericReporter</a></li>
	</ul>

	<li><a href="#genericarray">GenericArray</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_genericarray">Attributes of GenericArray</a></li>
		<li><a href="#associations_of_genericarray">Associations of GenericArray</a></li>
	</ul>

</ul>
<!-- INDEX END -->

<hr />
<p>
</p>
<hr />
<h1><a name="taxon">Taxon</a></h1>
<p>CaCORE::CaBIO::Taxon - Perl extension for Taxon.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Taxon is a Perl object representation of the
caBIO Taxon object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representing the various names (scientific, common, abbreviated, etc.) for a species associated with a specific instance of a Gene, Chromosome, Pathway, Protein, or Tissue.</p>
<p>
</p>
<h2><a name="attributes_of_taxon">Attributes of Taxon</a></h2>
<p>The following are all the attributes of the Taxon object and their data types:</p>
<dl>
<dt><strong><a name="item_scientificname">scientificName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_ethnicitystrain">ethnicityStrain</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_abbreviation">abbreviation</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_commonname">commonName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_id">id</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_taxon">Associations of Taxon</a></h2>
<p>The following are all the objects that are associated with the Taxon:</p>
<dl>
<dt><strong><a name="item__22gene_22_3a"><a href="#gene">Gene</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a name="item__22protein_22_3a"><a href="#protein">Protein</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getProteinCollection</code> to get a collection of associated Protein.</p>
</dd>
</li>
<dt><strong><a name="item__22pathway_22_3a"><a href="#pathway">Pathway</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getPathwayCollection</code> to get a collection of associated Pathway.</p>
</dd>
</li>
<dt><strong><a name="item__22tissue_22_3a"><a href="#tissue">Tissue</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getTissueCollection</code> to get a collection of associated Tissue.</p>
</dd>
</li>
<dt><strong><a name="item__22chromosome_22_3a"><a href="#chromosome">Chromosome</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getChromosomeCollection</code> to get a collection of associated Chromosome.</p>
</dd>
</li>
<dt><strong><a name="item__22clone_22_3a"><a href="#clone">Clone</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getCloneCollection</code> to get a collection of associated Clone.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="agent">Agent</a></h1>
<p>CaCORE::CaBIO::Agent - Perl extension for Agent.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Agent is a Perl object representation of the
caBIO Agent object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A therapeutic agent (drug, intervention therapy) used in a clinical trail protocol.</p>
<p>
</p>
<h2><a name="attributes_of_agent">Attributes of Agent</a></h2>
<p>The following are all the attributes of the Agent object and their data types:</p>
<dl>
<dt><strong><a name="item_iscmapagent">isCMAPAgent</a></strong>

<dd>
<p>data type: <code>boolean</code></p>
</dd>
</li>
<dt><strong><a name="item_comment">comment</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_source">source</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_nscnumber">NSCNumber</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_name">name</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_evsid">EVSId</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_agent">Associations of Agent</a></h2>
<p>The following are all the objects that are associated with the Agent:</p>
<dl>
<dt><strong><a name="item__22target_22_3a"><a href="#target">Target</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getTargetCollection</code> to get a collection of associated Target.</p>
</dd>
</li>
<dt><strong><a name="item__22clinicaltrialprotocol_22_3a"><a href="#clinicaltrialprotocol">ClinicalTrialProtocol</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getClinicalTrialProtocolCollection</code> to get a collection of associated ClinicalTrialProtocol.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="chromosome">Chromosome</a></h1>
<p>CaCORE::CaBIO::Chromosome - Perl extension for Chromosome.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Chromosome is a Perl object representation of the
caBIO Chromosome object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representing a specific chromosome for a specific taxon; provides access to all known genes contained in the chromosome and to the taxon.</p>
<p>
</p>
<h2><a name="attributes_of_chromosome">Attributes of Chromosome</a></h2>
<p>The following are all the attributes of the Chromosome object and their data types:</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_number">number</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_chromosome">Associations of Chromosome</a></h2>
<p>The following are all the objects that are associated with the Chromosome:</p>
<dl>
<dt><strong><a name="item__22taxon_22_3a"><a href="#taxon">Taxon</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getTaxon</code> to get the associated Taxon.</p>
</dd>
</li>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a name="item__22location_22_3a"><a href="#location">Location</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getLocationCollection</code> to get a collection of associated Location.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="gene">Gene</a></h1>
<p>CaCORE::CaBIO::Gene - Perl extension for Gene.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Gene is a Perl object representation of the
caBIO Gene object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Gene objects are the effective portal to most of the genomic information provided by the caBIO data services; organs, diseases, chromosomes, pathways, sequence data, and expression experiments are among the many objects accessible via a gene.</p>
<p>
</p>
<h2><a name="attributes_of_gene">Attributes of Gene</a></h2>
<p>The following are all the attributes of the Gene object and their data types:</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_clusterid">clusterId</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_symbol">symbol</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_fullname">fullName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_gene">Associations of Gene</a></h2>
<p>The following are all the objects that are associated with the Gene:</p>
<dl>
<dt><strong><a name="item__22geneontology_22_3a"><a href="#geneontology">GeneOntology</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGeneOntologyCollection</code> to get a collection of associated GeneOntology.</p>
</dd>
</li>
<dt><strong><a name="item__22genealias_22_3a"><a href="#genealias">GeneAlias</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGeneAliasCollection</code> to get a collection of associated GeneAlias.</p>
</dd>
</li>
<dt><strong><a href="#protein">Protein</a>:</strong>

<dd>
<p>One to many assoication, use <code>getProteinCollection</code> to get a collection of associated Protein.</p>
</dd>
</li>
<dt><strong><a name="item__22expressionfeature_22_3a"><a href="#expressionfeature">ExpressionFeature</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getExpressionFeatureCollection</code> to get a collection of associated ExpressionFeature.</p>
</dd>
</li>
<dt><strong><a href="#pathway">Pathway</a>:</strong>

<dd>
<p>One to many assoication, use <code>getPathwayCollection</code> to get a collection of associated Pathway.</p>
</dd>
</li>
<dt><strong><a name="item__22organontology_22_3a"><a href="#organontology">OrganOntology</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getOrganOntologyCollection</code> to get a collection of associated OrganOntology.</p>
</dd>
</li>
<dt><strong><a href="#taxon">Taxon</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getTaxon</code> to get the associated Taxon.</p>
</dd>
</li>
<dt><strong><a name="item__22library_22_3a"><a href="#library">Library</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getLibraryCollection</code> to get a collection of associated Library.</p>
</dd>
</li>
<dt><strong><a name="item__22generelativelocation_22_3a"><a href="#generelativelocation">GeneRelativeLocation</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGeneRelativeLocationCollection</code> to get a collection of associated GeneRelativeLocation.</p>
</dd>
</li>
<dt><strong><a href="#chromosome">Chromosome</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getChromosome</code> to get the associated Chromosome.</p>
</dd>
</li>
<dt><strong><a name="item__22databasecrossreference_22_3a"><a href="#databasecrossreference">DatabaseCrossReference</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getDatabaseCrossReferenceCollection</code> to get a collection of associated DatabaseCrossReference.</p>
</dd>
</li>
<dt><strong><a name="item__22genericreporter_22_3a"><a href="#genericreporter">GenericReporter</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGenericReporterCollection</code> to get a collection of associated GenericReporter.</p>
</dd>
</li>
<dt><strong><a href="#target">Target</a>:</strong>

<dd>
<p>One to many assoication, use <code>getTargetCollection</code> to get a collection of associated Target.</p>
</dd>
</li>
<dt><strong><a name="item__22histopathology_22_3a"><a href="#histopathology">Histopathology</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getHistopathologyCollection</code> to get a collection of associated Histopathology.</p>
</dd>
</li>
<dt><strong><a href="#location">Location</a>:</strong>

<dd>
<p>One to many assoication, use <code>getLocationCollection</code> to get a collection of associated Location.</p>
</dd>
</li>
<dt><strong><a name="item__22nucleicacidsequence_22_3a"><a href="#nucleicacidsequence">NucleicAcidSequence</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getNucleicAcidSequenceCollection</code> to get a collection of associated NucleicAcidSequence.</p>
</dd>
</li>
<dt><strong><a name="item__22homologousassociation_22_3a"><a href="#homologousassociation">HomologousAssociation</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getHomologousAssociationCollection</code> to get a collection of associated HomologousAssociation.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="vocabulary">Vocabulary</a></h1>
<p>CaCORE::CaBIO::Vocabulary - Perl extension for Vocabulary.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Vocabulary is a Perl object representation of the
caBIO Vocabulary object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>The object describes the vocabulary.</p>
<p>
</p>
<h2><a name="attributes_of_vocabulary">Attributes of Vocabulary</a></h2>
<p>The following are all the attributes of the Vocabulary object and their data types:</p>
<dl>
<dt><strong><a name="item_generalterm">generalTerm</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_coreterm">coreTerm</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_vocabulary">Associations of Vocabulary</a></h2>
<p>The following are all the objects that are associated with the Vocabulary:</p>
<dl>
<dt><strong><a name="item__22anomaly_22_3a"><a href="#anomaly">Anomaly</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getAnomalyCollection</code> to get a collection of associated Anomaly.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="protocol">Protocol</a></h1>
<p>CaCORE::CaBIO::Protocol - Perl extension for Protocol.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Protocol is a Perl object representation of the
caBIO Protocol object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representation of the protocol used in assembling a clone library.</p>
<p>
</p>
<h2><a name="attributes_of_protocol">Attributes of Protocol</a></h2>
<p>The following are all the attributes of the Protocol object and their data types:</p>
<dl>
<dt><strong><a name="item_type">type</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_description">description</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_protocol">Associations of Protocol</a></h2>
<p>The following are all the objects that are associated with the Protocol:</p>
<dl>
<dt><strong><a href="#library">Library</a>:</strong>

<dd>
<p>One to many assoication, use <code>getLibraryCollection</code> to get a collection of associated Library.</p>
</dd>
</li>
<dt><strong><a href="#tissue">Tissue</a>:</strong>

<dd>
<p>One to many assoication, use <code>getTissueCollection</code> to get a collection of associated Tissue.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="tissue">Tissue</a></h1>
<p>CaCORE::CaBIO::Tissue - Perl extension for Tissue.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Tissue is a Perl object representation of the
caBIO Tissue object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A group of similar cells united to perform a specific function.</p>
<p>
</p>
<h2><a name="attributes_of_tissue">Attributes of Tissue</a></h2>
<p>The following are all the attributes of the Tissue object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_celltype">cellType</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_histology">histology</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_sex">sex</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_cellline">cellLine</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_supplier">supplier</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_developmentalstage">developmentalStage</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_organ">organ</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_tissue">Associations of Tissue</a></h2>
<p>The following are all the objects that are associated with the Tissue:</p>
<dl>
<dt><strong><a name="item__22protocol_22_3a"><a href="#protocol">Protocol</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getProtocol</code> to get the associated Protocol.</p>
</dd>
</li>
<dt><strong><a href="#taxon">Taxon</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getTaxon</code> to get the associated Taxon.</p>
</dd>
</li>
<dt><strong><a href="#library">Library</a>:</strong>

<dd>
<p>One to many assoication, use <code>getLibraryCollection</code> to get a collection of associated Library.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="snp">SNP</a></h1>
<p>CaCORE::CaBIO::SNP - Perl extension for SNP.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::SNP is a Perl object representation of the
caBIO SNP object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representing a Single Nucleotide Polymorphism; provides access to the clones and the trace files from which it was identified, the two most common substitutions at that position, the offset of the SNP in the parent sequence, and a confidence score.</p>
<p>
</p>
<h2><a name="attributes_of_snp">Attributes of SNP</a></h2>
<p>The following are all the attributes of the SNP object and their data types:</p>
<dl>
<dt><strong><a name="item_allelea">alleleA</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_validationstatus">validationStatus</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_alleleb">alleleB</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_dbsnpid">DBSNPID</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_snp">Associations of SNP</a></h2>
<p>The following are all the objects that are associated with the SNP:</p>
<dl>
<dt><strong><a href="#databasecrossreference">DatabaseCrossReference</a>:</strong>

<dd>
<p>One to many assoication, use <code>getDatabaseCrossReferenceCollection</code> to get a collection of associated DatabaseCrossReference.</p>
</dd>
</li>
<dt><strong><a name="item__22populationfrequency_22_3a"><a href="#populationfrequency">PopulationFrequency</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getPopulationFrequencyCollection</code> to get a collection of associated PopulationFrequency.</p>
</dd>
</li>
<dt><strong><a href="#location">Location</a>:</strong>

<dd>
<p>One to many assoication, use <code>getLocationCollection</code> to get a collection of associated Location.</p>
</dd>
</li>
<dt><strong><a href="#generelativelocation">GeneRelativeLocation</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneRelativeLocationCollection</code> to get a collection of associated GeneRelativeLocation.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="genealias">GeneAlias</a></h1>
<p>CaCORE::CaBIO::GeneAlias - Perl extension for GeneAlias.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::GeneAlias is a Perl object representation of the
caBIO GeneAlias object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>This object describes the gene alias.</p>
<p>
</p>
<h2><a name="attributes_of_genealias">Attributes of GeneAlias</a></h2>
<p>The following are all the attributes of the GeneAlias object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_genealias">Associations of GeneAlias</a></h2>
<p>The following are all the objects that are associated with the GeneAlias:</p>
<dl>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="library">Library</a></h1>
<p>CaCORE::CaBIO::Library - Perl extension for Library.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Library is a Perl object representation of the
caBIO Library object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representing a CGAP library; provides access to information about: the tissue sample and its method of preparation, the library protocol that was used, the clones contained in the library, and the sequence information derived from the library.</p>
<p>
</p>
<h2><a name="attributes_of_library">Attributes of Library</a></h2>
<p>The following are all the attributes of the Library object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_rsite1">rsite1</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_creationdate">creationDate</a></strong>

<dd>
<p>data type: <code>dateTime</code></p>
</dd>
</li>
<dt><strong><a name="item_labhost">labHost</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_clonevector">cloneVector</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_clonestodate">clonesToDate</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_sequencestodate">sequencesToDate</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_keyword">keyword</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_cloneproducer">cloneProducer</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_rsite2">rsite2</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_clonevectortype">cloneVectorType</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_unigeneid">uniGeneId</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_library">Associations of Library</a></h2>
<p>The following are all the objects that are associated with the Library:</p>
<dl>
<dt><strong><a href="#protocol">Protocol</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getProtocol</code> to get the associated Protocol.</p>
</dd>
</li>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a href="#clone">Clone</a>:</strong>

<dd>
<p>One to many assoication, use <code>getCloneCollection</code> to get a collection of associated Clone.</p>
</dd>
</li>
<dt><strong><a href="#tissue">Tissue</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getTissue</code> to get the associated Tissue.</p>
</dd>
</li>
<dt><strong><a href="#histopathology">Histopathology</a>:</strong>

<dd>
<p>One to many assoication, use <code>getHistopathologyCollection</code> to get a collection of associated Histopathology.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="clone">Clone</a></h1>
<p>CaCORE::CaBIO::Clone - Perl extension for Clone.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Clone is a Perl object representation of the
caBIO Clone object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object used to hold information pertaining to I.M.A.G.E. clones; provides access to sequence information, associated trace files, and the clone's library.</p>
<p>
</p>
<h2><a name="attributes_of_clone">Attributes of Clone</a></h2>
<p>The following are all the attributes of the Clone object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_insertsize">insertSize</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_clone">Associations of Clone</a></h2>
<p>The following are all the objects that are associated with the Clone:</p>
<dl>
<dt><strong><a href="#taxon">Taxon</a>:</strong>

<dd>
<p>One to many assoication, use <code>getTaxonCollection</code> to get a collection of associated Taxon.</p>
</dd>
</li>
<dt><strong><a name="item__22clonerelativelocation_22_3a"><a href="#clonerelativelocation">CloneRelativeLocation</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getCloneRelativeLocationCollection</code> to get a collection of associated CloneRelativeLocation.</p>
</dd>
</li>
<dt><strong><a href="#library">Library</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getLibrary</code> to get the associated Library.</p>
</dd>
</li>
<dt><strong><a href="#nucleicacidsequence">NucleicAcidSequence</a>:</strong>

<dd>
<p>One to many assoication, use <code>getNucleicAcidSequenceCollection</code> to get a collection of associated NucleicAcidSequence.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="target">Target</a></h1>
<p>CaCORE::CaBIO::Target - Perl extension for Target.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Target is a Perl object representation of the
caBIO Target object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A gene thought to be at the root of a disease etiology, and which is targeted for therapeutic intervention in a clinical trial.</p>
<p>
</p>
<h2><a name="attributes_of_target">Attributes of Target</a></h2>
<p>The following are all the attributes of the Target object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_target">Associations of Target</a></h2>
<p>The following are all the objects that are associated with the Target:</p>
<dl>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a name="item__22agent_22_3a"><a href="#agent">Agent</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getAgentCollection</code> to get a collection of associated Agent.</p>
</dd>
</li>
<dt><strong><a href="#anomaly">Anomaly</a>:</strong>

<dd>
<p>One to many assoication, use <code>getAnomalyCollection</code> to get a collection of associated Anomaly.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="location">Location</a></h1>
<p>CaCORE::CaBIO::Location - Perl extension for Location.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Location is a Perl object representation of the
caBIO Location object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_location">Attributes of Location</a></h2>
<p>The following are all the attributes of the Location object and their data types:</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_location">Associations of Location</a></h2>
<p>The following are all the objects that are associated with the Location:</p>
<dl>
<dt><strong><a name="item__22snp_22_3a"><a href="#snp">SNP</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getSNP</code> to get the associated SNP.</p>
</dd>
</li>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getGene</code> to get the associated Gene.</p>
</dd>
</li>
<dt><strong><a href="#nucleicacidsequence">NucleicAcidSequence</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getNucleicAcidSequence</code> to get the associated NucleicAcidSequence.</p>
</dd>
</li>
<dt><strong><a href="#chromosome">Chromosome</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getChromosome</code> to get the associated Chromosome.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="clonerelativelocation">CloneRelativeLocation</a></h1>
<p>CaCORE::CaBIO::CloneRelativeLocation - Perl extension for CloneRelativeLocation.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::CloneRelativeLocation is a Perl object representation of the
caBIO CloneRelativeLocation object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_clonerelativelocation">Attributes of CloneRelativeLocation</a></h2>
<p>The following are all the attributes of the CloneRelativeLocation object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_clonerelativelocation">Associations of CloneRelativeLocation</a></h2>
<p>The following are all the objects that are associated with the CloneRelativeLocation:</p>
<dl>
<dt><strong><a href="#clone">Clone</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getClone</code> to get the associated Clone.</p>
</dd>
</li>
<dt><strong><a href="#nucleicacidsequence">NucleicAcidSequence</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getNucleicAcidSequence</code> to get the associated NucleicAcidSequence.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="nucleicacidsequence">NucleicAcidSequence</a></h1>
<p>CaCORE::CaBIO::NucleicAcidSequence - Perl extension for NucleicAcidSequence.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::NucleicAcidSequence is a Perl object representation of the
caBIO NucleicAcidSequence object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representation of a nucleic acid sequence; provides access to the clones from which it was derived.</p>
<p>
</p>
<h2><a name="attributes_of_nucleicacidsequence">Attributes of NucleicAcidSequence</a></h2>
<p>The following are all the attributes of the NucleicAcidSequence object and their data types:</p>
<dl>
<dt><strong><a name="item_value">value</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_length">length</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_accessionnumber">accessionNumber</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_accessionnumberversion">accessionNumberVersion</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_nucleicacidsequence">Associations of NucleicAcidSequence</a></h2>
<p>The following are all the objects that are associated with the NucleicAcidSequence:</p>
<dl>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a href="#databasecrossreference">DatabaseCrossReference</a>:</strong>

<dd>
<p>One to many assoication, use <code>getDatabaseCrossReferenceCollection</code> to get a collection of associated DatabaseCrossReference.</p>
</dd>
</li>
<dt><strong><a href="#location">Location</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getLocation</code> to get the associated Location.</p>
</dd>
</li>
<dt><strong><a href="#clonerelativelocation">CloneRelativeLocation</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getCloneRelativeLocation</code> to get the associated CloneRelativeLocation.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="diseaseontology">DiseaseOntology</a></h1>
<p>CaCORE::CaBIO::DiseaseOntology - Perl extension for DiseaseOntology.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::DiseaseOntology is a Perl object representation of the
caBIO DiseaseOntology object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Disease objects specify a disease name and ID; disease objects also provide access to: ontological relations to other diseases; clinical trial protocols treating the disease; and specific histologies associated with instances of the disease.</p>
<p>
</p>
<h2><a name="attributes_of_diseaseontology">Attributes of DiseaseOntology</a></h2>
<p>The following are all the attributes of the DiseaseOntology object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_diseaseontology">Associations of DiseaseOntology</a></h2>
<p>The following are all the objects that are associated with the DiseaseOntology:</p>
<dl>
<dt><strong><a name="item__22parentdiseaseontologyrelationship_22_3a"><a href="#parentdiseaseontologyrelationship">ParentDiseaseOntologyRelationship</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getParentDiseaseOntologyRelationshipCollection</code> to get a collection of associated ParentDiseaseOntologyRelationship.</p>
</dd>
</li>
<dt><strong><a name="item__22childdiseaseontologyrelationship_22_3a"><a href="#childdiseaseontologyrelationship">ChildDiseaseOntologyRelationship</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getChildDiseaseOntologyRelationshipCollection</code> to get a collection of associated ChildDiseaseOntologyRelationship.</p>
</dd>
</li>
<dt><strong><a href="#histopathology">Histopathology</a>:</strong>

<dd>
<p>One to many assoication, use <code>getHistopathologyCollection</code> to get a collection of associated Histopathology.</p>
</dd>
</li>
<dt><strong><a href="#clinicaltrialprotocol">ClinicalTrialProtocol</a>:</strong>

<dd>
<p>One to many assoication, use <code>getClinicalTrialProtocolCollection</code> to get a collection of associated ClinicalTrialProtocol.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="clinicaltrialprotocol">ClinicalTrialProtocol</a></h1>
<p>CaCORE::CaBIO::ClinicalTrialProtocol - Perl extension for ClinicalTrialProtocol.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::ClinicalTrialProtocol is a Perl object representation of the
caBIO ClinicalTrialProtocol object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>The protocol associated with a clinical trial; organizes administrative information about the trial such as Organization ID, participants, phase, etc., and provides access to the administered Agents.</p>
<p>
</p>
<h2><a name="attributes_of_clinicaltrialprotocol">Attributes of ClinicalTrialProtocol</a></h2>
<p>The following are all the attributes of the ClinicalTrialProtocol object and their data types:</p>
<dl>
<dt><strong><a name="item_currentstatus">currentStatus</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_participationtype">participationType</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_treatmentflag">treatmentFlag</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_leadorganizationid">leadOrganizationId</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_nihadmincode">NIHAdminCode</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_title">title</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_pdqidentifier">PDQIdentifier</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_documentnumber">documentNumber</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_leadorganizationname">leadOrganizationName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_phase">phase</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_piname">PIName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_currentstatusdate">currentStatusDate</a></strong>

<dd>
<p>data type: <code>dateTime</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_clinicaltrialprotocol">Associations of ClinicalTrialProtocol</a></h2>
<p>The following are all the objects that are associated with the ClinicalTrialProtocol:</p>
<dl>
<dt><strong><a href="#agent">Agent</a>:</strong>

<dd>
<p>One to many assoication, use <code>getAgentCollection</code> to get a collection of associated Agent.</p>
</dd>
</li>
<dt><strong><a name="item__22protocolassociation_22_3a"><a href="#protocolassociation">ProtocolAssociation</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getProtocolAssociationCollection</code> to get a collection of associated ProtocolAssociation.</p>
</dd>
</li>
<dt><strong><a name="item__22diseaseontology_22_3a"><a href="#diseaseontology">DiseaseOntology</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getDiseaseOntologyCollection</code> to get a collection of associated DiseaseOntology.</p>
</dd>
</li>
<dt><strong><a href="#histopathology">Histopathology</a>:</strong>

<dd>
<p>One to many assoication, use <code>getHistopathologyCollection</code> to get a collection of associated Histopathology.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="protocolassociation">ProtocolAssociation</a></h1>
<p>CaCORE::CaBIO::ProtocolAssociation - Perl extension for ProtocolAssociation.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::ProtocolAssociation is a Perl object representation of the
caBIO ProtocolAssociation object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An association class relating between Clinical Trial Protocols to Diseases.</p>
<p>
</p>
<h2><a name="attributes_of_protocolassociation">Attributes of ProtocolAssociation</a></h2>
<p>The following are all the attributes of the ProtocolAssociation object and their data types:</p>
<dl>
<dt><strong><a name="item_diseasesubcategory">diseaseSubCategory</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_ctepname">CTEPNAME</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_imtcode">IMTCODE</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_diseasecategory">diseaseCategory</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_protocolassociation">Associations of ProtocolAssociation</a></h2>
<p>The following are all the objects that are associated with the ProtocolAssociation:</p>
<dl>
<dt><strong><a href="#diseaseontology">DiseaseOntology</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getDiseaseOntology</code> to get the associated DiseaseOntology.</p>
</dd>
</li>
<dt><strong><a href="#clinicaltrialprotocol">ClinicalTrialProtocol</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getClinicalTrialProtocol</code> to get the associated ClinicalTrialProtocol.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="pathway">Pathway</a></h1>
<p>CaCORE::CaBIO::Pathway - Perl extension for Pathway.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Pathway is a Perl object representation of the
caBIO Pathway object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representation of a molecular/cellular pathway compiled by BioCarta. Pathways are associated with specific Taxon objects, and contain multiple Gene objects, which may be Targets for treatment.</p>
<p>
</p>
<h2><a name="attributes_of_pathway">Attributes of Pathway</a></h2>
<p>The following are all the attributes of the Pathway object and their data types:</p>
<dl>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_diagram">diagram</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_displayvalue">displayValue</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_pathway">Associations of Pathway</a></h2>
<p>The following are all the objects that are associated with the Pathway:</p>
<dl>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a href="#histopathology">Histopathology</a>:</strong>

<dd>
<p>One to many assoication, use <code>getHistopathologyCollection</code> to get a collection of associated Histopathology.</p>
</dd>
</li>
<dt><strong><a href="#taxon">Taxon</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getTaxon</code> to get the associated Taxon.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="organontology">OrganOntology</a></h1>
<p>CaCORE::CaBIO::OrganOntology - Perl extension for OrganOntology.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::OrganOntology is a Perl object representation of the
caBIO OrganOntology object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A representation of an organ whose name occurs in a controlled vocabulary; provides access to any Histopathology objects for the organ, and, because it is $quot;ontolog-able,$quot; provides access to its ancestral and descendant terms in the vocabulary.</p>
<p>
</p>
<h2><a name="attributes_of_organontology">Attributes of OrganOntology</a></h2>
<p>The following are all the attributes of the OrganOntology object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_organontology">Associations of OrganOntology</a></h2>
<p>The following are all the objects that are associated with the OrganOntology:</p>
<dl>
<dt><strong><a name="item__22parentorganontologyrelationship_22_3a"><a href="#parentorganontologyrelationship">ParentOrganOntologyRelationship</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getParentOrganOntologyRelationshipCollection</code> to get a collection of associated ParentOrganOntologyRelationship.</p>
</dd>
</li>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a href="#expressionfeature">ExpressionFeature</a>:</strong>

<dd>
<p>One to many assoication, use <code>getExpressionFeatureCollection</code> to get a collection of associated ExpressionFeature.</p>
</dd>
</li>
<dt><strong><a href="#histopathology">Histopathology</a>:</strong>

<dd>
<p>One to many assoication, use <code>getHistopathologyCollection</code> to get a collection of associated Histopathology.</p>
</dd>
</li>
<dt><strong><a name="item__22childorganontologyrelationship_22_3a"><a href="#childorganontologyrelationship">ChildOrganOntologyRelationship</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getChildOrganOntologyRelationshipCollection</code> to get a collection of associated ChildOrganOntologyRelationship.</p>
</dd>
</li>
<dt><strong><a href="#anomaly">Anomaly</a>:</strong>

<dd>
<p>One to many assoication, use <code>getAnomalyCollection</code> to get a collection of associated Anomaly.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="organontologyrelationship">OrganOntologyRelationship</a></h1>
<p>CaCORE::CaBIO::OrganOntologyRelationship - Perl extension for OrganOntologyRelationship.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::OrganOntologyRelationship is a Perl object representation of the
caBIO OrganOntologyRelationship object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Organ relationship object describes relationships among organs.</p>
<p>
</p>
<h2><a name="attributes_of_organontologyrelationship">Attributes of OrganOntologyRelationship</a></h2>
<p>The following are all the attributes of the OrganOntologyRelationship object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_organontologyrelationship">Associations of OrganOntologyRelationship</a></h2>
<p>The following are all the objects that are associated with the OrganOntologyRelationship:</p>
<dl>
<dt><strong><a name="item__22childorganontology_22_3a"><a href="#childorganontology">ChildOrganOntology</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getChildOrganOntology</code> to get the associated ChildOrganOntology.</p>
</dd>
</li>
<dt><strong><a name="item__22parentorganontology_22_3a"><a href="#parentorganontology">ParentOrganOntology</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getParentOrganOntology</code> to get the associated ParentOrganOntology.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="histopathology">Histopathology</a></h1>
<p>CaCORE::CaBIO::Histopathology - Perl extension for Histopathology.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Histopathology is a Perl object representation of the
caBIO Histopathology object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object representing anatomical changes in a diseased tissue sample associated with an expression experiment; captures the relationship between organ and disease.</p>
<p>
</p>
<h2><a name="attributes_of_histopathology">Attributes of Histopathology</a></h2>
<p>The following are all the attributes of the Histopathology object and their data types:</p>
<dl>
<dt><strong><a name="item_comments">comments</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_relationaloperation">relationalOperation</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_tumorincidencerate">tumorIncidenceRate</a></strong>

<dd>
<p>data type: <code>float</code></p>
</dd>
</li>
<dt><strong><a name="item_ageofonset">ageOfOnset</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_microscopicdescription">microscopicDescription</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_grossdescription">grossDescription</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_survivalinfo">survivalInfo</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_histopathology">Associations of Histopathology</a></h2>
<p>The following are all the objects that are associated with the Histopathology:</p>
<dl>
<dt><strong><a name="item__22metastasis_22_3a"><a href="#metastasis">Metastasis</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getMetastasisCollection</code> to get a collection of associated Metastasis.</p>
</dd>
</li>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a href="#library">Library</a>:</strong>

<dd>
<p>One to many assoication, use <code>getLibraryCollection</code> to get a collection of associated Library.</p>
</dd>
</li>
<dt><strong><a href="#diseaseontology">DiseaseOntology</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getDiseaseOntology</code> to get the associated DiseaseOntology.</p>
</dd>
</li>
<dt><strong><a href="#organontology">OrganOntology</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getOrganOntology</code> to get the associated OrganOntology.</p>
</dd>
</li>
<dt><strong><a href="#clinicaltrialprotocol">ClinicalTrialProtocol</a>:</strong>

<dd>
<p>One to many assoication, use <code>getClinicalTrialProtocolCollection</code> to get a collection of associated ClinicalTrialProtocol.</p>
</dd>
</li>
<dt><strong><a href="#anomaly">Anomaly</a>:</strong>

<dd>
<p>One to many assoication, use <code>getAnomalyCollection</code> to get a collection of associated Anomaly.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="geneontology">GeneOntology</a></h1>
<p>CaCORE::CaBIO::GeneOntology - Perl extension for GeneOntology.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::GeneOntology is a Perl object representation of the
caBIO GeneOntology object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An object providing entry to a Gene object's position in the Gene Ontology Consortium's controlled vocabularies, as recorded by LocusLink; provides access to gene objects corresponding to the ontological term, as well as to ancestor and descendant terms in the ontology tree.</p>
<p>
</p>
<h2><a name="attributes_of_geneontology">Attributes of GeneOntology</a></h2>
<p>The following are all the attributes of the GeneOntology object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_geneontology">Associations of GeneOntology</a></h2>
<p>The following are all the objects that are associated with the GeneOntology:</p>
<dl>
<dt><strong><a name="item__22childgeneontologyrelationship_22_3a"><a href="#childgeneontologyrelationship">ChildGeneOntologyRelationship</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getChildGeneOntologyRelationshipCollection</code> to get a collection of associated ChildGeneOntologyRelationship.</p>
</dd>
</li>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a name="item__22parentgeneontologyrelationship_22_3a"><a href="#parentgeneontologyrelationship">ParentGeneOntologyRelationship</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getParentGeneOntologyRelationshipCollection</code> to get a collection of associated ParentGeneOntologyRelationship.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="geneontologyrelationship">GeneOntologyRelationship</a></h1>
<p>CaCORE::CaBIO::GeneOntologyRelationship - Perl extension for GeneOntologyRelationship.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::GeneOntologyRelationship is a Perl object representation of the
caBIO GeneOntologyRelationship object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>This object specifies GoOntologyrRelationship.</p>
<p>
</p>
<h2><a name="attributes_of_geneontologyrelationship">Attributes of GeneOntologyRelationship</a></h2>
<p>The following are all the attributes of the GeneOntologyRelationship object and their data types:</p>
<dl>
<dt><strong><a name="item_relationshiptype">relationshipType</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_geneontologyrelationship">Associations of GeneOntologyRelationship</a></h2>
<p>The following are all the objects that are associated with the GeneOntologyRelationship:</p>
<dl>
<dt><strong><a name="item__22parentgeneontology_22_3a"><a href="#parentgeneontology">ParentGeneOntology</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getParentGeneOntology</code> to get the associated ParentGeneOntology.</p>
</dd>
</li>
<dt><strong><a name="item__22childgeneontology_22_3a"><a href="#childgeneontology">ChildGeneOntology</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getChildGeneOntology</code> to get the associated ChildGeneOntology.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="diseaseontologyrelationship">DiseaseOntologyRelationship</a></h1>
<p>CaCORE::CaBIO::DiseaseOntologyRelationship - Perl extension for DiseaseOntologyRelationship.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::DiseaseOntologyRelationship is a Perl object representation of the
caBIO DiseaseOntologyRelationship object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>The diseaser relationship specifies the relationship among diseases.</p>
<p>
</p>
<h2><a name="attributes_of_diseaseontologyrelationship">Attributes of DiseaseOntologyRelationship</a></h2>
<p>The following are all the attributes of the DiseaseOntologyRelationship object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_diseaseontologyrelationship">Associations of DiseaseOntologyRelationship</a></h2>
<p>The following are all the objects that are associated with the DiseaseOntologyRelationship:</p>
<dl>
<dt><strong><a name="item__22childdiseaseontology_22_3a"><a href="#childdiseaseontology">ChildDiseaseOntology</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getChildDiseaseOntology</code> to get the associated ChildDiseaseOntology.</p>
</dd>
</li>
<dt><strong><a name="item__22parentdiseaseontology_22_3a"><a href="#parentdiseaseontology">ParentDiseaseOntology</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getParentDiseaseOntology</code> to get the associated ParentDiseaseOntology.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="anomaly">Anomaly</a></h1>
<p>CaCORE::CaBIO::Anomaly - Perl extension for Anomaly.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Anomaly is a Perl object representation of the
caBIO Anomaly object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An irregularity in either the expression of a gene or its structure (i.e., a mutation).</p>
<p>
</p>
<h2><a name="attributes_of_anomaly">Attributes of Anomaly</a></h2>
<p>The following are all the attributes of the Anomaly object and their data types:</p>
<dl>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_anomaly">Associations of Anomaly</a></h2>
<p>The following are all the objects that are associated with the Anomaly:</p>
<dl>
<dt><strong><a href="#organontology">OrganOntology</a>:</strong>

<dd>
<p>One to many assoication, use <code>getOrganOntologyCollection</code> to get a collection of associated OrganOntology.</p>
</dd>
</li>
<dt><strong><a href="#histopathology">Histopathology</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getHistopathology</code> to get the associated Histopathology.</p>
</dd>
</li>
<dt><strong><a name="item__22vocabulary_22_3a"><a href="#vocabulary">Vocabulary</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getVocabularyCollection</code> to get a collection of associated Vocabulary.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="populationfrequency">PopulationFrequency</a></h1>
<p>CaCORE::CaBIO::PopulationFrequency - Perl extension for PopulationFrequency.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::PopulationFrequency is a Perl object representation of the
caBIO PopulationFrequency object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_populationfrequency">Attributes of PopulationFrequency</a></h2>
<p>The following are all the attributes of the PopulationFrequency object and their data types:</p>
<dl>
<dt><strong><a name="item_majorfrequency">majorFrequency</a></strong>

<dd>
<p>data type: <code>double</code></p>
</dd>
</li>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_minorallele">minorAllele</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_ethnicity">ethnicity</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_minorfrequency">minorFrequency</a></strong>

<dd>
<p>data type: <code>double</code></p>
</dd>
</li>
<dt><strong><a name="item_majorallele">majorAllele</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_populationfrequency">Associations of PopulationFrequency</a></h2>
<p>The following are all the objects that are associated with the PopulationFrequency:</p>
<dl>
<dt><strong><a href="#snp">SNP</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getSNP</code> to get the associated SNP.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="proteinsequence">ProteinSequence</a></h1>
<p>CaCORE::CaBIO::ProteinSequence - Perl extension for ProteinSequence.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::ProteinSequence is a Perl object representation of the
caBIO ProteinSequence object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_proteinsequence">Attributes of ProteinSequence</a></h2>
<p>The following are all the attributes of the ProteinSequence object and their data types:

</p>
<dl>
<dt><strong><a name="item_checksum">checkSum</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>value</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_molecularweightindaltons">molecularWeightInDaltons</a></strong>

<dd>
<p>data type: <code>double</code>

</p>
</dd>
</li>
<dt><strong>length</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_proteinsequence">Associations of ProteinSequence</a></h2>
<p>The following are all the objects that are associated with the ProteinSequence:

</p>
<dl>
<dt><strong><a href="#protein">Protein</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getProtein</code> to get the associated Protein.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="protein">Protein</a></h1>
<p>CaCORE::CaBIO::Protein - Perl extension for Protein.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Protein is a Perl object representation of the
caBIO Protein object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_protein">Attributes of Protein</a></h2>
<p>The following are all the attributes of the Protein object and their data types:

</p>
<dl>
<dt><strong><a name="item_copyrightstatement">copyrightStatement</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_primaryaccession">primaryAccession</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_uniprotcode">uniProtCode</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_protein">Associations of Protein</a></h2>
<p>The following are all the objects that are associated with the Protein:

</p>
<dl>
<dt><strong><a href="#taxon">Taxon</a>:</strong>

<dd>
<p>One to many assoication, use <code>getTaxonCollection</code> to get a collection of associated Taxon.

</p>
</dd>
</li>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.

</p>
</dd>
</li>
<dt><strong><a name="item__22secondaryaccession_22_3a"><a href="#secondaryaccession">SecondaryAccession</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getSecondaryAccessionCollection</code> to get a collection of associated SecondaryAccession.

</p>
</dd>
</li>
<dt><strong><a name="item__22keywords_22_3a"><a href="#keywords">Keywords</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getKeywordsCollection</code> to get a collection of associated Keywords.

</p>
</dd>
</li>
<dt><strong><a name="item__22proteinsequence_22_3a"><a href="#proteinsequence">ProteinSequence</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getProteinSequence</code> to get the associated ProteinSequence.

</p>
</dd>
</li>
<dt><strong><a name="item__22proteinalias_22_3a"><a href="#proteinalias">ProteinAlias</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getProteinAliasCollection</code> to get a collection of associated ProteinAlias.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="proteinalias">ProteinAlias</a></h1>
<p>CaCORE::CaBIO::ProteinAlias - Perl extension for ProteinAlias.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::ProteinAlias is a Perl object representation of the
caBIO ProteinAlias object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_proteinalias">Attributes of ProteinAlias</a></h2>
<p>The following are all the attributes of the ProteinAlias object and their data types:

</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_proteinalias">Associations of ProteinAlias</a></h2>
<p>The following are all the objects that are associated with the ProteinAlias:

</p>
<dl>
<dt><strong><a href="#protein">Protein</a>:</strong>

<dd>
<p>One to many assoication, use <code>getProteinCollection</code> to get a collection of associated Protein.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="homologousassociation">HomologousAssociation</a></h1>
<p>CaCORE::CaBIO::HomologousAssociation - Perl extension for HomologousAssociation.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::HomologousAssociation is a Perl object representation of the
caBIO HomologousAssociation object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_homologousassociation">Attributes of HomologousAssociation</a></h2>
<p>The following are all the attributes of the HomologousAssociation object and their data types:

</p>
<dl>
<dt><strong><a name="item_similaritypercentage">similarityPercentage</a></strong>

<dd>
<p>data type: <code>float</code>

</p>
</dd>
</li>
<dt><strong><a name="item_homologousid">homologousId</a></strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_homologousassociation">Associations of HomologousAssociation</a></h2>
<p>The following are all the objects that are associated with the HomologousAssociation:

</p>
<dl>
<dt><strong><a name="item__22homologousgene_22_3a"><a href="#homologousgene">HomologousGene</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getHomologousGene</code> to get the associated HomologousGene.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="physicallocation">PhysicalLocation</a></h1>
<p>CaCORE::CaBIO::PhysicalLocation - Perl extension for PhysicalLocation.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::PhysicalLocation is a Perl object representation of the
caBIO PhysicalLocation object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_physicallocation">Attributes of PhysicalLocation</a></h2>
<p>The following are all the attributes of the PhysicalLocation object and their data types:

</p>
<dl>
<dt><strong><a name="item_chromosomalstartposition">chromosomalStartPosition</a></strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
</li>
<dt><strong><a name="item_chromosomalendposition">chromosomalEndPosition</a></strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_physicallocation">Associations of PhysicalLocation</a></h2>
<p>The following are all the objects that are associated with the PhysicalLocation:

</p>
<dl>
<dt><strong><a name="item__22cytoband_22_3a"><a href="#cytoband">Cytoband</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getCytobandCollection</code> to get a collection of associated Cytoband.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="cytoband">Cytoband</a></h1>
<p>CaCORE::CaBIO::Cytoband - Perl extension for Cytoband.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::Cytoband is a Perl object representation of the
caBIO Cytoband object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_cytoband">Attributes of Cytoband</a></h2>
<p>The following are all the attributes of the Cytoband object and their data types:

</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_cytoband">Associations of Cytoband</a></h2>
<p>The following are all the objects that are associated with the Cytoband:

</p>
<dl>
<dt><strong><a name="item__22physicallocation_22_3a"><a href="#physicallocation">PhysicalLocation</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getPhysicalLocation</code> to get the associated PhysicalLocation.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="cytogeneticlocation">CytogeneticLocation</a></h1>
<p>CaCORE::CaBIO::CytogeneticLocation - Perl extension for CytogeneticLocation.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::CytogeneticLocation is a Perl object representation of the
caBIO CytogeneticLocation object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_cytogeneticlocation">Attributes of CytogeneticLocation</a></h2>
<p>The following are all the attributes of the CytogeneticLocation object and their data types:

</p>
<pre>
  End Attributes

</pre>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_cytogeneticlocation">Associations of CytogeneticLocation</a></h2>
<p>The following are all the objects that are associated with the CytogeneticLocation:

</p>
<dl>
<dt><strong><a name="item__22startcytoband_22_3a"><a href="#startcytoband">StartCytoband</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getStartCytoband</code> to get the associated StartCytoband.

</p>
</dd>
</li>
<dt><strong><a name="item__22endcytoband_22_3a"><a href="#endcytoband">EndCytoband</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getEndCytoband</code> to get the associated EndCytoband.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="generelativelocation">GeneRelativeLocation</a></h1>
<p>CaCORE::CaBIO::GeneRelativeLocation - Perl extension for GeneRelativeLocation.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::GeneRelativeLocation is a Perl object representation of the
caBIO GeneRelativeLocation object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_generelativelocation">Attributes of GeneRelativeLocation</a></h2>
<p>The following are all the attributes of the GeneRelativeLocation object and their data types:

</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_generelativelocation">Associations of GeneRelativeLocation</a></h2>
<p>The following are all the objects that are associated with the GeneRelativeLocation:

</p>
<pre>
  End Associations and related methods

</pre>
<p>
</p>
<hr />
<h1><a name="genericreporter">GenericReporter</a></h1>
<p>CaCORE::CaBIO::GenericReporter - Perl extension for GenericReporter.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::GenericReporter is a Perl object representation of the
caBIO GenericReporter object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_genericreporter">Attributes of GenericReporter</a></h2>
<p>The following are all the attributes of the GenericReporter object and their data types:

</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_genericreporter">Associations of GenericReporter</a></h2>
<p>The following are all the objects that are associated with the GenericReporter:

</p>
<dl>
<dt><strong><a name="item__22genericarray_22_3a"><a href="#genericarray">GenericArray</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGenericArrayCollection</code> to get a collection of associated GenericArray.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="genericarray">GenericArray</a></h1>
<p>CaCORE::CaBIO::GenericArray - Perl extension for GenericArray.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaBIO::GenericArray is a Perl object representation of the
caBIO GenericArray object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_genericarray">Attributes of GenericArray</a></h2>
<p>The following are all the attributes of the GenericArray object and their data types:

</p>
<dl>
<dt><strong><a name="item_arrayname">arrayName</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_platform">platform</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_genericarray">Associations of GenericArray</a></h2>
<p>The following are all the objects that are associated with the GenericArray:

</p>
<dl>
<dt><strong><a href="#genericreporter">GenericReporter</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGenericReporterCollection</code> to get a collection of associated GenericReporter.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>

</body>

</html>
