<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>CaCORE::CaMOD</title>
<link rev="made" href="mailto:" />
</head>

<body style="background-color: white">

<p><a name="__index__"></a></p>
<!-- INDEX BEGIN -->

<ul>

	<li><a href="#approvalstatus">ApprovalStatus</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_approvalstatus">Attributes of ApprovalStatus</a></li>
		<li><a href="#associations_of_approvalstatus">Associations of ApprovalStatus</a></li>
	</ul>

	<li><a href="#availability">Availability</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_availability">Attributes of Availability</a></li>
		<li><a href="#associations_of_availability">Associations of Availability</a></li>
	</ul>

	<li><a href="#abstractcancermodel">AbstractCancerModel</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_abstractcancermodel">Attributes of AbstractCancerModel</a></li>
		<li><a href="#associations_of_abstractcancermodel">Associations of AbstractCancerModel</a></li>
	</ul>

	<li><a href="#tumorcode">TumorCode</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_tumorcode">Attributes of TumorCode</a></li>
		<li><a href="#associations_of_tumorcode">Associations of TumorCode</a></li>
	</ul>

	<li><a href="#xenograft">Xenograft</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_xenograft">Attributes of Xenograft</a></li>
		<li><a href="#associations_of_xenograft">Associations of Xenograft</a></li>
	</ul>

	<li><a href="#treatmentschedule">TreatmentSchedule</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_treatmentschedule">Attributes of TreatmentSchedule</a></li>
		<li><a href="#associations_of_treatmentschedule">Associations of TreatmentSchedule</a></li>
	</ul>

	<li><a href="#nomenclature">Nomenclature</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_nomenclature">Attributes of Nomenclature</a></li>
		<li><a href="#associations_of_nomenclature">Associations of Nomenclature</a></li>
	</ul>

	<li><a href="#genotypesummary">GenotypeSummary</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_genotypesummary">Attributes of GenotypeSummary</a></li>
		<li><a href="#associations_of_genotypesummary">Associations of GenotypeSummary</a></li>
	</ul>

	<li><a href="#conditionality">Conditionality</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_conditionality">Attributes of Conditionality</a></li>
		<li><a href="#associations_of_conditionality">Associations of Conditionality</a></li>
	</ul>

	<li><a href="#image">Image</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_image">Attributes of Image</a></li>
		<li><a href="#associations_of_image">Associations of Image</a></li>
	</ul>

	<li><a href="#engineeredgene">EngineeredGene</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_engineeredgene">Attributes of EngineeredGene</a></li>
		<li><a href="#associations_of_engineeredgene">Associations of EngineeredGene</a></li>
	</ul>

	<li><a href="#integrationtype">IntegrationType</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_integrationtype">Attributes of IntegrationType</a></li>
		<li><a href="#associations_of_integrationtype">Associations of IntegrationType</a></li>
	</ul>

	<li><a href="#transgene">Transgene</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_transgene">Attributes of Transgene</a></li>
		<li><a href="#associations_of_transgene">Associations of Transgene</a></li>
	</ul>

	<li><a href="#therapy">Therapy</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_therapy">Attributes of Therapy</a></li>
		<li><a href="#associations_of_therapy">Associations of Therapy</a></li>
	</ul>

	<li><a href="#yeastmodel">YeastModel</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_yeastmodel">Attributes of YeastModel</a></li>
		<li><a href="#associations_of_yeastmodel">Associations of YeastModel</a></li>
	</ul>

	<li><a href="#targetedmodification">TargetedModification</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_targetedmodification">Attributes of TargetedModification</a></li>
		<li><a href="#associations_of_targetedmodification">Associations of TargetedModification</a></li>
	</ul>

	<li><a href="#sexdistribution">SexDistribution</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_sexdistribution">Attributes of SexDistribution</a></li>
		<li><a href="#associations_of_sexdistribution">Associations of SexDistribution</a></li>
	</ul>

	<li><a href="#segmenttype">SegmentType</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_segmenttype">Attributes of SegmentType</a></li>
		<li><a href="#associations_of_segmenttype">Associations of SegmentType</a></li>
	</ul>

	<li><a href="#role">Role</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_role">Attributes of Role</a></li>
		<li><a href="#associations_of_role">Associations of Role</a></li>
	</ul>

	<li><a href="#repositoryinfo">RepositoryInfo</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_repositoryinfo">Attributes of RepositoryInfo</a></li>
		<li><a href="#associations_of_repositoryinfo">Associations of RepositoryInfo</a></li>
	</ul>

	<li><a href="#regulatoryelementtype">RegulatoryElementType</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_regulatoryelementtype">Attributes of RegulatoryElementType</a></li>
		<li><a href="#associations_of_regulatoryelementtype">Associations of RegulatoryElementType</a></li>
	</ul>

	<li><a href="#regulatoryelement">RegulatoryElement</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_regulatoryelement">Attributes of RegulatoryElement</a></li>
		<li><a href="#associations_of_regulatoryelement">Associations of RegulatoryElement</a></li>
	</ul>

	<li><a href="#publicationstatus">PublicationStatus</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_publicationstatus">Attributes of PublicationStatus</a></li>
		<li><a href="#associations_of_publicationstatus">Associations of PublicationStatus</a></li>
	</ul>

	<li><a href="#publication">Publication</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_publication">Attributes of Publication</a></li>
		<li><a href="#associations_of_publication">Associations of Publication</a></li>
	</ul>

	<li><a href="#promoter">Promoter</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_promoter">Attributes of Promoter</a></li>
		<li><a href="#associations_of_promoter">Associations of Promoter</a></li>
	</ul>

	<li><a href="#phenotype">Phenotype</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_phenotype">Attributes of Phenotype</a></li>
		<li><a href="#associations_of_phenotype">Associations of Phenotype</a></li>
	</ul>

	<li><a href="#party">Party</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_party">Attributes of Party</a></li>
		<li><a href="#associations_of_party">Associations of Party</a></li>
	</ul>

	<li><a href="#person">Person</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_person">Attributes of Person</a></li>
		<li><a href="#associations_of_person">Associations of Person</a></li>
	</ul>

	<li><a href="#partyrole">PartyRole</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_partyrole">Attributes of PartyRole</a></li>
		<li><a href="#associations_of_partyrole">Associations of PartyRole</a></li>
	</ul>

	<li><a href="#organization">Organization</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_organization">Attributes of Organization</a></li>
		<li><a href="#associations_of_organization">Associations of Organization</a></li>
	</ul>

	<li><a href="#microarraydata">MicroArrayData</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_microarraydata">Attributes of MicroArrayData</a></li>
		<li><a href="#associations_of_microarraydata">Associations of MicroArrayData</a></li>
	</ul>

	<li><a href="#jaxinfo">JaxInfo</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_jaxinfo">Attributes of JaxInfo</a></li>
		<li><a href="#associations_of_jaxinfo">Associations of JaxInfo</a></li>
	</ul>

	<li><a href="#environmentalfactor">EnvironmentalFactor</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_environmentalfactor">Attributes of EnvironmentalFactor</a></li>
		<li><a href="#associations_of_environmentalfactor">Associations of EnvironmentalFactor</a></li>
	</ul>

	<li><a href="#inducedmutation">InducedMutation</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_inducedmutation">Attributes of InducedMutation</a></li>
		<li><a href="#associations_of_inducedmutation">Associations of InducedMutation</a></li>
	</ul>

	<li><a href="#genomicsegment">GenomicSegment</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_genomicsegment">Attributes of GenomicSegment</a></li>
		<li><a href="#associations_of_genomicsegment">Associations of GenomicSegment</a></li>
	</ul>

	<li><a href="#geneticalteration">GeneticAlteration</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_geneticalteration">Attributes of GeneticAlteration</a></li>
		<li><a href="#associations_of_geneticalteration">Associations of GeneticAlteration</a></li>
	</ul>

	<li><a href="#genefunction">GeneFunction</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_genefunction">Attributes of GeneFunction</a></li>
		<li><a href="#associations_of_genefunction">Associations of GeneFunction</a></li>
	</ul>

	<li><a href="#genedelivery">GeneDelivery</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_genedelivery">Attributes of GeneDelivery</a></li>
		<li><a href="#associations_of_genedelivery">Associations of GeneDelivery</a></li>
	</ul>

	<li><a href="#contactinfo">ContactInfo</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_contactinfo">Attributes of ContactInfo</a></li>
		<li><a href="#associations_of_contactinfo">Associations of ContactInfo</a></li>
	</ul>

	<li><a href="#cellline">CellLine</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_cellline">Attributes of CellLine</a></li>
		<li><a href="#associations_of_cellline">Associations of CellLine</a></li>
	</ul>

	<li><a href="#carcinogenicintervention">CarcinogenicIntervention</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_carcinogenicintervention">Attributes of CarcinogenicIntervention</a></li>
		<li><a href="#associations_of_carcinogenicintervention">Associations of CarcinogenicIntervention</a></li>
	</ul>

	<li><a href="#animalmodel">AnimalModel</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_animalmodel">Attributes of AnimalModel</a></li>
		<li><a href="#associations_of_animalmodel">Associations of AnimalModel</a></li>
	</ul>

	<li><a href="#endpointcode">EndpointCode</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_endpointcode">Attributes of EndpointCode</a></li>
		<li><a href="#associations_of_endpointcode">Associations of EndpointCode</a></li>
	</ul>

	<li><a href="#screeningresult">ScreeningResult</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_screeningresult">Attributes of ScreeningResult</a></li>
		<li><a href="#associations_of_screeningresult">Associations of ScreeningResult</a></li>
	</ul>

	<li><a href="#invivoresult">InvivoResult</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_invivoresult">Attributes of InvivoResult</a></li>
		<li><a href="#associations_of_invivoresult">Associations of InvivoResult</a></li>
	</ul>

	<li><a href="#modificationtype">ModificationType</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_modificationtype">Attributes of ModificationType</a></li>
		<li><a href="#associations_of_modificationtype">Associations of ModificationType</a></li>
	</ul>

	<li><a href="#expressionfeature">ExpressionFeature</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_expressionfeature">Attributes of ExpressionFeature</a></li>
		<li><a href="#associations_of_expressionfeature">Associations of ExpressionFeature</a></li>
	</ul>

	<li><a href="#cancermodel">CancerModel</a></li>
	<ul>

		<li><a href="#abstract">Abstract</a></li>
		<li><a href="#description">Description</a></li>
		<li><a href="#attributes_of_cancermodel">Attributes of CancerModel</a></li>
		<li><a href="#associations_of_cancermodel">Associations of CancerModel</a></li>
	</ul>

</ul>
<!-- INDEX END -->

<hr />
<p>
</p>
<hr />
<h1><a name="approvalstatus">ApprovalStatus</a></h1>
<p>CaCORE::CaMOD::ApprovalStatus - Perl extension for ApprovalStatus.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::ApprovalStatus is a Perl object representation of the
caBIO ApprovalStatus object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_approvalstatus">Attributes of ApprovalStatus</a></h2>
<p>The following are all the attributes of the ApprovalStatus object and their data types:</p>
<dl>
<dt><strong><a name="item_name">name</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_id">id</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_approvalstatus">Associations of ApprovalStatus</a></h2>
<p>The following are all the objects that are associated with the ApprovalStatus:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="availability">Availability</a></h1>
<p>CaCORE::CaMOD::Availability - Perl extension for Availability.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Availability is a Perl object representation of the
caBIO Availability object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Dates on which a animal model record was entered, modified or release to the intended viewing audience.</p>
<p>
</p>
<h2><a name="attributes_of_availability">Attributes of Availability</a></h2>
<p>The following are all the attributes of the Availability object and their data types:</p>
<dl>
<dt><strong><a name="item_entereddate">enteredDate</a></strong>

<dd>
<p>data type: <code>dateTime</code></p>
</dd>
</li>
<dt><strong><a name="item_visibleto">visibleTo</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_releasedate">releaseDate</a></strong>

<dd>
<p>data type: <code>dateTime</code></p>
</dd>
</li>
<dt><strong><a name="item_modifieddate">modifiedDate</a></strong>

<dd>
<p>data type: <code>dateTime</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_availability">Associations of Availability</a></h2>
<p>The following are all the objects that are associated with the Availability:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="abstractcancermodel">AbstractCancerModel</a></h1>
<p>CaCORE::CaMOD::AbstractCancerModel - Perl extension for AbstractCancerModel.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::AbstractCancerModel is a Perl object representation of the
caBIO AbstractCancerModel object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_abstractcancermodel">Attributes of AbstractCancerModel</a></h2>
<p>The following are all the attributes of the AbstractCancerModel object and their data types:</p>
<dl>
<dt><strong><a name="item_experimentdesign">experimentDesign</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_modeldescriptor">modelDescriptor</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_abstractcancermodel">Associations of AbstractCancerModel</a></h2>
<p>The following are all the objects that are associated with the AbstractCancerModel:</p>
<dl>
<dt><strong><a name="item__22approvalstatus_22_3a"><a href="#approvalstatus">ApprovalStatus</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getApprovalStatus</code> to get the associated ApprovalStatus.</p>
</dd>
</li>
<dt><strong><a name="item__22availability_22_3a"><a href="#availability">Availability</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getAvailability</code> to get the associated Availability.</p>
</dd>
</li>
<dt><strong><a name="item__22partyrole_22_3a"><a href="#partyrole">PartyRole</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getPartyRoleCollection</code> to get a collection of associated PartyRole.</p>
</dd>
</li>
<dt><strong><a name="item__22publication_22_3a"><a href="#publication">Publication</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getPublicationCollection</code> to get a collection of associated Publication.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="tumorcode">TumorCode</a></h1>
<p>CaCORE::CaMOD::TumorCode - Perl extension for TumorCode.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::TumorCode is a Perl object representation of the
caBIO TumorCode object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A tumor code captures the tumor type and origin species used for a drug screen test.</p>
<p>
</p>
<h2><a name="attributes_of_tumorcode">Attributes of TumorCode</a></h2>
<p>The following are all the attributes of the TumorCode object and their data types:</p>
<dl>
<dt><strong><a name="item_code">code</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_description">description</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_tumorcode">Associations of TumorCode</a></h2>
<p>The following are all the objects that are associated with the TumorCode:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="xenograft">Xenograft</a></h1>
<p>CaCORE::CaMOD::Xenograft - Perl extension for Xenograft.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Xenograft is a Perl object representation of the
caBIO Xenograft object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>This object describes to xenografts (implanting tissue or cell lines from a different species) and allografts (implanting tissue or cell lines form the same species). The grafts are transplanted in immuno-compromised animals where they grow and, depending on the graft material, form tumors.</p>
<p>
</p>
<h2><a name="attributes_of_xenograft">Attributes of Xenograft</a></h2>
<p>The following are all the attributes of the Xenograft object and their data types:</p>
<dl>
<dt><strong><a name="item_geneticmanipulation">geneticManipulation</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_administrativesite">administrativeSite</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_type">type</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_parentalcelllinename">parentalCellLineName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_modificationdescription">modificationDescription</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_xenograft">Associations of Xenograft</a></h2>
<p>The following are all the objects that are associated with the Xenograft:</p>
<dl>
<dt><strong><a name="item__22invivoresult_22_3a"><a href="#invivoresult">InvivoResult</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getInvivoResultCollection</code> to get a collection of associated InvivoResult.</p>
</dd>
</li>
<dt><strong><a name="item__22tumorcode_22_3a"><a href="#tumorcode">TumorCode</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getTumorCode</code> to get the associated TumorCode.</p>
</dd>
</li>
<dt><strong><a name="item__22origintaxon_22_3a"><a href="#origintaxon">OriginTaxon</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getOriginTaxon</code> to get the associated OriginTaxon.</p>
</dd>
</li>
<dt><strong><a name="item__22hosttaxon_22_3a"><a href="#hosttaxon">HostTaxon</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getHostTaxon</code> to get the associated HostTaxon.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="treatmentschedule">TreatmentSchedule</a></h1>
<p>CaCORE::CaMOD::TreatmentSchedule - Perl extension for TreatmentSchedule.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::TreatmentSchedule is a Perl object representation of the
caBIO TreatmentSchedule object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Treatment schedule for the pre-clinical trial / therapeutic experiment describing the dosages of the drug and the times of the treatment.</p>
<p>
</p>
<h2><a name="attributes_of_treatmentschedule">Attributes of TreatmentSchedule</a></h2>
<p>The following are all the attributes of the TreatmentSchedule object and their data types:</p>
<dl>
<dt><strong><a name="item_regimen">regimen</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_dosage">dosage</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_treatmentschedule">Associations of TreatmentSchedule</a></h2>
<p>The following are all the objects that are associated with the TreatmentSchedule:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="nomenclature">Nomenclature</a></h1>
<p>CaCORE::CaMOD::Nomenclature - Perl extension for Nomenclature.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Nomenclature is a Perl object representation of the
caBIO Nomenclature object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Official nomenclature name for mouse and rat models.</p>
<p>
</p>
<h2><a name="attributes_of_nomenclature">Attributes of Nomenclature</a></h2>
<p>The following are all the attributes of the Nomenclature object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_nomenclature">Associations of Nomenclature</a></h2>
<p>The following are all the objects that are associated with the Nomenclature:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="genotypesummary">GenotypeSummary</a></h1>
<p>CaCORE::CaMOD::GenotypeSummary - Perl extension for GenotypeSummary.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::GenotypeSummary is a Perl object representation of the
caBIO GenotypeSummary object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Listing of the genetic changes made deliberately in order to generate the animal model.</p>
<p>
</p>
<h2><a name="attributes_of_genotypesummary">Attributes of GenotypeSummary</a></h2>
<p>The following are all the attributes of the GenotypeSummary object and their data types:</p>
<dl>
<dt><strong><a name="item_summary">summary</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_genotype">genotype</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_genotypesummary">Associations of GenotypeSummary</a></h2>
<p>The following are all the objects that are associated with the GenotypeSummary:</p>
<dl>
<dt><strong><a name="item__22nomenclature_22_3a"><a href="#nomenclature">Nomenclature</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getNomenclature</code> to get the associated Nomenclature.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="conditionality">Conditionality</a></h1>
<p>CaCORE::CaMOD::Conditionality - Perl extension for Conditionality.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Conditionality is a Perl object representation of the
caBIO Conditionality object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Indicates if a transgene or targeted modification is done conditionally (time or tissue specific).</p>
<p>
</p>
<h2><a name="attributes_of_conditionality">Attributes of Conditionality</a></h2>
<p>The following are all the attributes of the Conditionality object and their data types:</p>
<dl>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_conditionedby">conditionedBy</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_conditionality">Associations of Conditionality</a></h2>
<p>The following are all the objects that are associated with the Conditionality:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="image">Image</a></h1>
<p>CaCORE::CaMOD::Image - Perl extension for Image.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Image is a Perl object representation of the
caBIO Image object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Images related to the animal model e.g. histology images, blots, and graphics.</p>
<p>
</p>
<h2><a name="attributes_of_image">Attributes of Image</a></h2>
<p>The following are all the attributes of the Image object and their data types:</p>
<dl>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_title">title</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_staining">staining</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_image">Associations of Image</a></h2>
<p>The following are all the objects that are associated with the Image:</p>
<dl>
<dt><strong><a href="#availability">Availability</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getAvailability</code> to get the associated Availability.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="engineeredgene">EngineeredGene</a></h1>
<p>CaCORE::CaMOD::EngineeredGene - Perl extension for EngineeredGene.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::EngineeredGene is a Perl object representation of the
caBIO EngineeredGene object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Manipulated Gene introduced in the animal model.</p>
<p>
</p>
<h2><a name="attributes_of_engineeredgene">Attributes of EngineeredGene</a></h2>
<p>The following are all the attributes of the EngineeredGene object and their data types:</p>
<dl>
<dt><strong><a name="item_cabioid">cabioId</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_engineeredgene">Associations of EngineeredGene</a></h2>
<p>The following are all the objects that are associated with the EngineeredGene:</p>
<dl>
<dt><strong><a name="item__22gene_22_3a"><a href="#gene">Gene</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGeneCollection</code> to get a collection of associated Gene.</p>
</dd>
</li>
<dt><strong><a name="item__22databasecrossreference_22_3a"><a href="#databasecrossreference">DatabaseCrossReference</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getDatabaseCrossReferenceCollection</code> to get a collection of associated DatabaseCrossReference.</p>
</dd>
</li>
<dt><strong><a name="item__22genotypesummary_22_3a"><a href="#genotypesummary">GenotypeSummary</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getGenotypeSummary</code> to get the associated GenotypeSummary.</p>
</dd>
</li>
<dt><strong><a name="item__22expressionfeature_22_3a"><a href="#expressionfeature">ExpressionFeature</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getExpressionFeatureCollection</code> to get a collection of associated ExpressionFeature.</p>
</dd>
</li>
<dt><strong><a name="item__22conditionality_22_3a"><a href="#conditionality">Conditionality</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getConditionality</code> to get the associated Conditionality.</p>
</dd>
</li>
<dt><strong><a name="item__22image_22_3a"><a href="#image">Image</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getImage</code> to get the associated Image.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="integrationtype">IntegrationType</a></h1>
<p>CaCORE::CaMOD::IntegrationType - Perl extension for IntegrationType.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::IntegrationType is a Perl object representation of the
caBIO IntegrationType object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Location of the integration of the engineered gene e.g. random or targeted.</p>
<p>
</p>
<h2><a name="attributes_of_integrationtype">Attributes of IntegrationType</a></h2>
<p>The following are all the attributes of the IntegrationType object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_integrationtype">Associations of IntegrationType</a></h2>
<p>The following are all the objects that are associated with the IntegrationType:</p>
<dl>
<dt><strong><a name="item__22genomicsegment_22_3a"><a href="#genomicsegment">GenomicSegment</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getGenomicSegmentCollection</code> to get a collection of associated GenomicSegment.</p>
</dd>
</li>
<dt><strong><a name="item__22transgene_22_3a"><a href="#transgene">Transgene</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getTransgeneCollection</code> to get a collection of associated Transgene.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="transgene">Transgene</a></h1>
<p>CaCORE::CaMOD::Transgene - Perl extension for Transgene.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Transgene is a Perl object representation of the
caBIO Transgene object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A transgene is an engineered gene consisting at least of a regulatory element and a coding sequence; regulatory element and coding sequence can come from different species.</p>
<p>
</p>
<h2><a name="attributes_of_transgene">Attributes of Transgene</a></h2>
<p>The following are all the attributes of the Transgene object and their data types:</p>
<dl>
<dt><strong><a name="item_locationofintegration">locationOfIntegration</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_engineeredgeneid">engineeredGeneId</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_transgene">Associations of Transgene</a></h2>
<p>The following are all the objects that are associated with the Transgene:</p>
<dl>
<dt><strong><a name="item__22regulatoryelement_22_3a"><a href="#regulatoryelement">RegulatoryElement</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getRegulatoryElementCollection</code> to get a collection of associated RegulatoryElement.</p>
</dd>
</li>
<dt><strong><a name="item__22integrationtype_22_3a"><a href="#integrationtype">IntegrationType</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getIntegrationType</code> to get the associated IntegrationType.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="therapy">Therapy</a></h1>
<p>CaCORE::CaMOD::Therapy - Perl extension for Therapy.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Therapy is a Perl object representation of the
caBIO Therapy object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Pre-clinical trials experiment conducted on the animal model with the goal to reduce cancer burden.</p>
<p>
</p>
<h2><a name="attributes_of_therapy">Attributes of Therapy</a></h2>
<p>The following are all the attributes of the Therapy object and their data types:</p>
<dl>
<dt><strong><a name="item_experiment">experiment</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_comments">comments</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_results">results</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_therapy">Associations of Therapy</a></h2>
<p>The following are all the objects that are associated with the Therapy:</p>
<dl>
<dt><strong><a name="item__22treatmentschedule_22_3a"><a href="#treatmentschedule">TreatmentSchedule</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getTreatmentSchedule</code> to get the associated TreatmentSchedule.</p>
</dd>
</li>
<dt><strong><a name="item__22agent_22_3a"><a href="#agent">Agent</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getAgent</code> to get the associated Agent.</p>
</dd>
</li>
<dt><strong><a href="#publication">Publication</a>:</strong>

<dd>
<p>One to many assoication, use <code>getPublicationCollection</code> to get a collection of associated Publication.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="yeastmodel">YeastModel</a></h1>
<p>CaCORE::CaMOD::YeastModel - Perl extension for YeastModel.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::YeastModel is a Perl object representation of the
caBIO YeastModel object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A yeast model captures the yeast strains altered in the NCI Yeast Anticancer Drug Screen.</p>
<p>
</p>
<h2><a name="attributes_of_yeastmodel">Attributes of YeastModel</a></h2>
<p>The following are all the attributes of the YeastModel object and their data types:</p>
<pre>
  End Attributes</pre>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_yeastmodel">Associations of YeastModel</a></h2>
<p>The following are all the objects that are associated with the YeastModel:</p>
<dl>
<dt><strong><a name="item__22targetedmodification_22_3a"><a href="#targetedmodification">TargetedModification</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getTargetedModificationCollection</code> to get a collection of associated TargetedModification.</p>
</dd>
</li>
<dt><strong><a name="item__22screeningresult_22_3a"><a href="#screeningresult">ScreeningResult</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getScreeningResultCollection</code> to get a collection of associated ScreeningResult.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="targetedmodification">TargetedModification</a></h1>
<p>CaCORE::CaMOD::TargetedModification - Perl extension for TargetedModification.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::TargetedModification is a Perl object representation of the
caBIO TargetedModification object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Modification targeted to a specific gene or a specific part of the gene to either suppress the function of the gene or insert marker sequences like loxP sites.</p>
<p>
</p>
<h2><a name="attributes_of_targetedmodification">Attributes of TargetedModification</a></h2>
<p>The following are all the attributes of the TargetedModification object and their data types:</p>
<dl>
<dt><strong><a name="item_escelllinename">esCellLineName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_blastocystname">blastocystName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_targetedmodification">Associations of TargetedModification</a></h2>
<p>The following are all the objects that are associated with the TargetedModification:</p>
<dl>
<dt><strong><a name="item__22modificationtype_22_3a"><a href="#modificationtype">ModificationType</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getModificationTypeCollection</code> to get a collection of associated ModificationType.</p>
</dd>
</li>
<dt><strong><a name="item__22yeastmodel_22_3a"><a href="#yeastmodel">YeastModel</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getYeastModel</code> to get the associated YeastModel.</p>
</dd>
</li>
<dt><strong><a name="item__22engineeredgene_22_3a"><a href="#engineeredgene">EngineeredGene</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getEngineeredGene</code> to get the associated EngineeredGene.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="sexdistribution">SexDistribution</a></h1>
<p>CaCORE::CaMOD::SexDistribution - Perl extension for SexDistribution.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::SexDistribution is a Perl object representation of the
caBIO SexDistribution object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Object indicated if the phenotype was observed in both sexes or only in one.</p>
<p>
</p>
<h2><a name="attributes_of_sexdistribution">Attributes of SexDistribution</a></h2>
<p>The following are all the attributes of the SexDistribution object and their data types:</p>
<dl>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_sexdistribution">Associations of SexDistribution</a></h2>
<p>The following are all the objects that are associated with the SexDistribution:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="segmenttype">SegmentType</a></h1>
<p>CaCORE::CaMOD::SegmentType - Perl extension for SegmentType.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::SegmentType is a Perl object representation of the
caBIO SegmentType object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>specifies the type of library that the genomic segment was retrieved from such as BAC or YAC.</p>
<p>
</p>
<h2><a name="attributes_of_segmenttype">Attributes of SegmentType</a></h2>
<p>The following are all the attributes of the SegmentType object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_segmenttype">Associations of SegmentType</a></h2>
<p>The following are all the objects that are associated with the SegmentType:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="role">Role</a></h1>
<p>CaCORE::CaMOD::Role - Perl extension for Role.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Role is a Perl object representation of the
caBIO Role object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Role that a person or organization plays e.g. public user, reviewer.</p>
<p>
</p>
<h2><a name="attributes_of_role">Attributes of Role</a></h2>
<p>The following are all the attributes of the Role object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_role">Associations of Role</a></h2>
<p>The following are all the objects that are associated with the Role:</p>
<dl>
<dt><strong><a href="#partyrole">PartyRole</a>:</strong>

<dd>
<p>One to many assoication, use <code>getPartyRoleCollection</code> to get a collection of associated PartyRole.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="repositoryinfo">RepositoryInfo</a></h1>
<p>CaCORE::CaMOD::RepositoryInfo - Perl extension for RepositoryInfo.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::RepositoryInfo is a Perl object representation of the
caBIO RepositoryInfo object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Some animal models are distributed by the MMHCC mouse repository. Object contains information about the availability of a particular model from the repository. A submitter of data to the cancer models database can also indicate if their model should be submitted to the repository.</p>
<p>
</p>
<h2><a name="attributes_of_repositoryinfo">Attributes of RepositoryInfo</a></h2>
<p>The following are all the attributes of the RepositoryInfo object and their data types:</p>
<dl>
<dt><strong><a name="item_intherepository">inTheRepository</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_sentemailcontent">sentEmailContent</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_suggestsubmission">suggestSubmission</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_repositoryinfo">Associations of RepositoryInfo</a></h2>
<p>The following are all the objects that are associated with the RepositoryInfo:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="regulatoryelementtype">RegulatoryElementType</a></h1>
<p>CaCORE::CaMOD::RegulatoryElementType - Perl extension for RegulatoryElementType.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::RegulatoryElementType is a Perl object representation of the
caBIO RegulatoryElementType object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Type of the regulatory element e.g. promoter, polyA signal.</p>
<p>
</p>
<h2><a name="attributes_of_regulatoryelementtype">Attributes of RegulatoryElementType</a></h2>
<p>The following are all the attributes of the RegulatoryElementType object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_regulatoryelementtype">Associations of RegulatoryElementType</a></h2>
<p>The following are all the objects that are associated with the RegulatoryElementType:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="regulatoryelement">RegulatoryElement</a></h1>
<p>CaCORE::CaMOD::RegulatoryElement - Perl extension for RegulatoryElement.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::RegulatoryElement is a Perl object representation of the
caBIO RegulatoryElement object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A regulatory element controls the expression of a gene and/or and engineered gene.</p>
<p>
</p>
<h2><a name="attributes_of_regulatoryelement">Attributes of RegulatoryElement</a></h2>
<p>The following are all the attributes of the RegulatoryElement object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_regulatoryelement">Associations of RegulatoryElement</a></h2>
<p>The following are all the objects that are associated with the RegulatoryElement:</p>
<dl>
<dt><strong><a name="item__22promoter_22_3a"><a href="#promoter">Promoter</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getPromoterCollection</code> to get a collection of associated Promoter.</p>
</dd>
</li>
<dt><strong><a name="item__22taxon_22_3a"><a href="#taxon">Taxon</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getTaxon</code> to get the associated Taxon.</p>
</dd>
</li>
<dt><strong><a name="item__22regulatoryelementtype_22_3a"><a href="#regulatoryelementtype">RegulatoryElementType</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getRegulatoryElementType</code> to get the associated RegulatoryElementType.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="publicationstatus">PublicationStatus</a></h1>
<p>CaCORE::CaMOD::PublicationStatus - Perl extension for PublicationStatus.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::PublicationStatus is a Perl object representation of the
caBIO PublicationStatus object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Status of the publication e.g. published, submitted.</p>
<p>
</p>
<h2><a name="attributes_of_publicationstatus">Attributes of PublicationStatus</a></h2>
<p>The following are all the attributes of the PublicationStatus object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_publicationstatus">Associations of PublicationStatus</a></h2>
<p>The following are all the objects that are associated with the PublicationStatus:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="publication">Publication</a></h1>
<p>CaCORE::CaMOD::Publication - Perl extension for Publication.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Publication is a Perl object representation of the
caBIO Publication object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Publications describing the animal model itself or experiments in which the animal model was used.</p>
<p>
</p>
<h2><a name="attributes_of_publication">Attributes of Publication</a></h2>
<p>The following are all the attributes of the Publication object and their data types:</p>
<dl>
<dt><strong><a name="item_volume">volume</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_endpage">endPage</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_year">year</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong>title</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_pmid">pmid</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_startpage">startPage</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_journal">journal</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_authors">authors</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_publication">Associations of Publication</a></h2>
<p>The following are all the objects that are associated with the Publication:</p>
<dl>
<dt><strong><a name="item__22publicationstatus_22_3a"><a href="#publicationstatus">PublicationStatus</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getPublicationStatus</code> to get the associated PublicationStatus.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="promoter">Promoter</a></h1>
<p>CaCORE::CaMOD::Promoter - Perl extension for Promoter.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Promoter is a Perl object representation of the
caBIO Promoter object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Promoter is a subcategory of the regulatory element.</p>
<p>
</p>
<h2><a name="attributes_of_promoter">Attributes of Promoter</a></h2>
<p>The following are all the attributes of the Promoter object and their data types:</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_promoter">Associations of Promoter</a></h2>
<p>The following are all the objects that are associated with the Promoter:</p>
<dl>
<dt><strong><a href="#gene">Gene</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getGene</code> to get the associated Gene.</p>
</dd>
</li>
<dt><strong><a href="#regulatoryelement">RegulatoryElement</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getRegulatoryElement</code> to get the associated RegulatoryElement.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="phenotype">Phenotype</a></h1>
<p>CaCORE::CaMOD::Phenotype - Perl extension for Phenotype.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Phenotype is a Perl object representation of the
caBIO Phenotype object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Phenotype displayed by the animal model such as neoplastic lesions, other diseases, behavioral problem.</p>
<p>
</p>
<h2><a name="attributes_of_phenotype">Attributes of Phenotype</a></h2>
<p>The following are all the attributes of the Phenotype object and their data types:</p>
<dl>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_breedingnotes">breedingNotes</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_phenotype">Associations of Phenotype</a></h2>
<p>The following are all the objects that are associated with the Phenotype:</p>
<dl>
<dt><strong><a name="item__22sexdistribution_22_3a"><a href="#sexdistribution">SexDistribution</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getSexDistribution</code> to get the associated SexDistribution.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="party">Party</a></h1>
<p>CaCORE::CaMOD::Party - Perl extension for Party.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Party is a Perl object representation of the
caBIO Party object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Entity that has access to the data, either a person or an organization.</p>
<p>
</p>
<h2><a name="attributes_of_party">Attributes of Party</a></h2>
<p>The following are all the attributes of the Party object and their data types:</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_party">Associations of Party</a></h2>
<p>The following are all the objects that are associated with the Party:</p>
<dl>
<dt><strong><a href="#partyrole">PartyRole</a>:</strong>

<dd>
<p>One to many assoication, use <code>getPartyRoleCollection</code> to get a collection of associated PartyRole.</p>
</dd>
</li>
<dt><strong><a name="item__22contactinfo_22_3a"><a href="#contactinfo">ContactInfo</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getContactInfoCollection</code> to get a collection of associated ContactInfo.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="person">Person</a></h1>
<p>CaCORE::CaMOD::Person - Perl extension for Person.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Person is a Perl object representation of the
caBIO Person object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Person's first and last name and the connection to the party.</p>
<p>
</p>
<h2><a name="attributes_of_person">Attributes of Person</a></h2>
<p>The following are all the attributes of the Person object and their data types:</p>
<dl>
<dt><strong><a name="item_partyid">partyId</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_firstname">firstName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_lastname">lastName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_person">Associations of Person</a></h2>
<p>The following are all the objects that are associated with the Person:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="partyrole">PartyRole</a></h1>
<p>CaCORE::CaMOD::PartyRole - Perl extension for PartyRole.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::PartyRole is a Perl object representation of the
caBIO PartyRole object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Object connects the party (organization or person) with the animal model.</p>
<p>
</p>
<h2><a name="attributes_of_partyrole">Attributes of PartyRole</a></h2>
<p>The following are all the attributes of the PartyRole object and their data types:</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_partyrole">Associations of PartyRole</a></h2>
<p>The following are all the objects that are associated with the PartyRole:</p>
<dl>
<dt><strong><a name="item__22role_22_3a"><a href="#role">Role</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getRole</code> to get the associated Role.</p>
</dd>
</li>
<dt><strong><a name="item__22animalmodel_22_3a"><a href="#animalmodel">AnimalModel</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getAnimalModelCollection</code> to get a collection of associated AnimalModel.</p>
</dd>
</li>
<dt><strong><a name="item__22party_22_3a"><a href="#party">Party</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getParty</code> to get the associated Party.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="organization">Organization</a></h1>
<p>CaCORE::CaMOD::Organization - Perl extension for Organization.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::Organization is a Perl object representation of the
caBIO Organization object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Organizational unit like a laboratory, institute or consortium.</p>
<p>
</p>
<h2><a name="attributes_of_organization">Attributes of Organization</a></h2>
<p>The following are all the attributes of the Organization object and their data types:</p>
<dl>
<dt><strong>partyId</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_organization">Associations of Organization</a></h2>
<p>The following are all the objects that are associated with the Organization:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="microarraydata">MicroArrayData</a></h1>
<p>CaCORE::CaMOD::MicroArrayData - Perl extension for MicroArrayData.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::MicroArrayData is a Perl object representation of the
caBIO MicroArrayData object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Data of microarray experiments generated from the animal model.</p>
<p>
</p>
<h2><a name="attributes_of_microarraydata">Attributes of MicroArrayData</a></h2>
<p>The following are all the attributes of the MicroArrayData object and their data types:</p>
<dl>
<dt><strong><a name="item_experimentname">experimentName</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_experimentid">experimentId</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_microarraydata">Associations of MicroArrayData</a></h2>
<p>The following are all the objects that are associated with the MicroArrayData:</p>
<dl>
<dt><strong><a href="#availability">Availability</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getAvailability</code> to get the associated Availability.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="jaxinfo">JaxInfo</a></h1>
<p>CaCORE::CaMOD::JaxInfo - Perl extension for JaxInfo.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::JaxInfo is a Perl object representation of the
caBIO JaxInfo object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Identification number (stock number) of strain at the Jackson Laboratory, number can be used for ordering the strain; <a href="http://www.jax.org.">http://www.jax.org.</a></p>
<p>
</p>
<h2><a name="attributes_of_jaxinfo">Attributes of JaxInfo</a></h2>
<p>The following are all the attributes of the JaxInfo object and their data types:</p>
<dl>
<dt><strong><a name="item_jaxstockno">jaxStockNo</a></strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_jaxinfo">Associations of JaxInfo</a></h2>
<p>The following are all the objects that are associated with the JaxInfo:</p>
<pre>
  End Associations and related methods</pre>
<p>
</p>
<hr />
<h1><a name="environmentalfactor">EnvironmentalFactor</a></h1>
<p>CaCORE::CaMOD::EnvironmentalFactor - Perl extension for EnvironmentalFactor.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::EnvironmentalFactor is a Perl object representation of the
caBIO EnvironmentalFactor object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Chemical, radiation, hormone treatment or other environmental factor that initiates or supports development of neoplasias.</p>
<p>
</p>
<h2><a name="attributes_of_environmentalfactor">Attributes of EnvironmentalFactor</a></h2>
<p>The following are all the attributes of the EnvironmentalFactor object and their data types:</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>type</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_environmentalfactor">Associations of EnvironmentalFactor</a></h2>
<p>The following are all the objects that are associated with the EnvironmentalFactor:</p>
<dl>
<dt><strong><a name="item__22inducedmutation_22_3a"><a href="#inducedmutation">InducedMutation</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getInducedMutationCollection</code> to get a collection of associated InducedMutation.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="inducedmutation">InducedMutation</a></h1>
<p>CaCORE::CaMOD::InducedMutation - Perl extension for InducedMutation.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::InducedMutation is a Perl object representation of the
caBIO InducedMutation object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Induced mutations are defined as mutations triggered by radiation, chemicals or other means. Progeny of the treated animal inherits the mutation.</p>
<p>
</p>
<h2><a name="attributes_of_inducedmutation">Attributes of InducedMutation</a></h2>
<p>The following are all the attributes of the InducedMutation object and their data types:</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_inducedmutation">Associations of InducedMutation</a></h2>
<p>The following are all the objects that are associated with the InducedMutation:</p>
<dl>
<dt><strong><a name="item__22environmentalfactor_22_3a"><a href="#environmentalfactor">EnvironmentalFactor</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getEnvironmentalFactor</code> to get the associated EnvironmentalFactor.</p>
</dd>
</li>
<dt><strong><a href="#engineeredgene">EngineeredGene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getEngineeredGeneCollection</code> to get a collection of associated EngineeredGene.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="genomicsegment">GenomicSegment</a></h1>
<p>CaCORE::CaMOD::GenomicSegment - Perl extension for GenomicSegment.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::GenomicSegment is a Perl object representation of the
caBIO GenomicSegment object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Genomic segment extracted from a library e.g. BAC or YAC library to be used in the genetic manipulation of the animal model.</p>
<p>
</p>
<h2><a name="attributes_of_genomicsegment">Attributes of GenomicSegment</a></h2>
<p>The following are all the attributes of the GenomicSegment object and their data types:</p>
<dl>
<dt><strong>locationOfIntegration</strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_segmentsize">segmentSize</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
</li>
<dt><strong><a name="item_clonedesignator">cloneDesignator</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_genomicsegment">Associations of GenomicSegment</a></h2>
<p>The following are all the objects that are associated with the GenomicSegment:</p>
<dl>
<dt><strong><a name="item__22segmenttype_22_3a"><a href="#segmenttype">SegmentType</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getSegmentType</code> to get the associated SegmentType.</p>
</dd>
</li>
<dt><strong><a href="#engineeredgene">EngineeredGene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getEngineeredGeneCollection</code> to get a collection of associated EngineeredGene.</p>
</dd>
<dd>
<pre>
  End Associations and related methods</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="geneticalteration">GeneticAlteration</a></h1>
<p>CaCORE::CaMOD::GeneticAlteration - Perl extension for GeneticAlteration.</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::GeneticAlteration is a Perl object representation of the
caBIO GeneticAlteration object.</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Genetic alterations found in the neoplastic lesions of the animal model. These alterations are not made intentionally.</p>
<p>
</p>
<h2><a name="attributes_of_geneticalteration">Attributes of GeneticAlteration</a></h2>
<p>The following are all the attributes of the GeneticAlteration object and their data types:</p>
<dl>
<dt><strong><a name="item_observation">observation</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong><a name="item_methodofobservation">methodOfObservation</a></strong>

<dd>
<p>data type: <code>string</code></p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code></p>
</dd>
<dd>
<pre>
  End Attributes</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.</pre>
<p>
</p>
<h2><a name="associations_of_geneticalteration">Associations of GeneticAlteration</a></h2>
<p>The following are all the objects that are associated with the GeneticAlteration:

</p>
<pre>
  End Associations and related methods

</pre>
<p>
</p>
<hr />
<h1><a name="genefunction">GeneFunction</a></h1>
<p>CaCORE::CaMOD::GeneFunction - Perl extension for GeneFunction.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::GeneFunction is a Perl object representation of the
caBIO GeneFunction object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Functions of the engineered genes.

</p>
<p>
</p>
<h2><a name="attributes_of_genefunction">Attributes of GeneFunction</a></h2>
<p>The following are all the attributes of the GeneFunction object and their data types:

</p>
<dl>
<dt><strong><a name="item_function">function</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_genefunction">Associations of GeneFunction</a></h2>
<p>The following are all the objects that are associated with the GeneFunction:

</p>
<dl>
<dt><strong><a href="#engineeredgene">EngineeredGene</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getEngineeredGene</code> to get the associated EngineeredGene.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="genedelivery">GeneDelivery</a></h1>
<p>CaCORE::CaMOD::GeneDelivery - Perl extension for GeneDelivery.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::GeneDelivery is a Perl object representation of the
caBIO GeneDelivery object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Gene is delivered to specific organs or specific receptors within the animal model. using viral vectors

</p>
<p>
</p>
<h2><a name="attributes_of_genedelivery">Attributes of GeneDelivery</a></h2>
<p>The following are all the attributes of the GeneDelivery object and their data types:

</p>
<dl>
<dt><strong><a name="item_viralvector">viralVector</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_genedelivery">Associations of GeneDelivery</a></h2>
<p>The following are all the objects that are associated with the GeneDelivery:

</p>
<dl>
<dt><strong><a href="#engineeredgene">EngineeredGene</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getEngineeredGene</code> to get the associated EngineeredGene.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="contactinfo">ContactInfo</a></h1>
<p>CaCORE::CaMOD::ContactInfo - Perl extension for ContactInfo.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::ContactInfo is a Perl object representation of the
caBIO ContactInfo object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Information regarding the person who submitted the data.

</p>
<p>
</p>
<h2><a name="attributes_of_contactinfo">Attributes of ContactInfo</a></h2>
<p>The following are all the attributes of the ContactInfo object and their data types:

</p>
<dl>
<dt><strong><a name="item_faxnumber">faxNumber</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_zipcode">zipCode</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_street">street</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_institute">institute</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_state">state</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_email">email</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_phonenumber">phoneNumber</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_labname">labName</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
</li>
<dt><strong><a name="item_city">city</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_contactinfo">Associations of ContactInfo</a></h2>
<p>The following are all the objects that are associated with the ContactInfo:

</p>
<dl>
<dt><strong><a href="#party">Party</a>:</strong>

<dd>
<p>One to many assoication, use <code>getPartyCollection</code> to get a collection of associated Party.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="cellline">CellLine</a></h1>
<p>CaCORE::CaMOD::CellLine - Perl extension for CellLine.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::CellLine is a Perl object representation of the
caBIO CellLine object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Cell lines generated from the animal model.

</p>
<p>
</p>
<h2><a name="attributes_of_cellline">Attributes of CellLine</a></h2>
<p>The following are all the attributes of the CellLine object and their data types:

</p>
<dl>
<dt><strong>experiment</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>comments</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>results</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_cellline">Associations of CellLine</a></h2>
<p>The following are all the objects that are associated with the CellLine:

</p>
<dl>
<dt><strong><a href="#publication">Publication</a>:</strong>

<dd>
<p>One to many assoication, use <code>getPublicationCollection</code> to get a collection of associated Publication.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="carcinogenicintervention">CarcinogenicIntervention</a></h1>
<p>CaCORE::CaMOD::CarcinogenicIntervention - Perl extension for CarcinogenicIntervention.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::CarcinogenicIntervention is a Perl object representation of the
caBIO CarcinogenicIntervention object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>Treatment or procedure that the animal model was exposed to in order to initiate or support the development of neoplastic lesions.

</p>
<p>
</p>
<h2><a name="attributes_of_carcinogenicintervention">Attributes of CarcinogenicIntervention</a></h2>
<p>The following are all the attributes of the CarcinogenicIntervention object and their data types:

</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_carcinogenicintervention">Associations of CarcinogenicIntervention</a></h2>
<p>The following are all the objects that are associated with the CarcinogenicIntervention:

</p>
<dl>
<dt><strong><a href="#environmentalfactor">EnvironmentalFactor</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getEnvironmentalFactor</code> to get the associated EnvironmentalFactor.

</p>
</dd>
</li>
<dt><strong><a href="#treatmentschedule">TreatmentSchedule</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getTreatmentSchedule</code> to get the associated TreatmentSchedule.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="animalmodel">AnimalModel</a></h1>
<p>CaCORE::CaMOD::AnimalModel - Perl extension for AnimalModel.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::AnimalModel is a Perl object representation of the
caBIO AnimalModel object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An animal model which develops cancer or can be used to generate a model that develops cancer.

</p>
<p>
</p>
<h2><a name="attributes_of_animalmodel">Attributes of AnimalModel</a></h2>
<p>The following are all the attributes of the AnimalModel object and their data types:

</p>
<dl>
<dt><strong><a name="item_url">url</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_animalmodel">Associations of AnimalModel</a></h2>
<p>The following are all the objects that are associated with the AnimalModel:

</p>
<dl>
<dt><strong><a name="item__22phenotype_22_3a"><a href="#phenotype">Phenotype</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getPhenotype</code> to get the associated Phenotype.

</p>
</dd>
</li>
<dt><strong><a name="item__22carcinogenicintervention_22_3a"><a href="#carcinogenicintervention">CarcinogenicIntervention</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getCarcinogenicInterventionCollection</code> to get a collection of associated CarcinogenicIntervention.

</p>
</dd>
</li>
<dt><strong><a name="item__22therapy_22_3a"><a href="#therapy">Therapy</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getTherapyCollection</code> to get a collection of associated Therapy.

</p>
</dd>
</li>
<dt><strong><a href="#image">Image</a>:</strong>

<dd>
<p>One to many assoication, use <code>getImageCollection</code> to get a collection of associated Image.

</p>
</dd>
</li>
<dt><strong><a name="item__22xenograft_22_3a"><a href="#xenograft">Xenograft</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getXenograftCollection</code> to get a collection of associated Xenograft.

</p>
</dd>
</li>
<dt><strong><a name="item__22microarraydata_22_3a"><a href="#microarraydata">MicroArrayData</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getMicroArrayDataCollection</code> to get a collection of associated MicroArrayData.

</p>
</dd>
</li>
<dt><strong><a href="#targetedmodification">TargetedModification</a>:</strong>

<dd>
<p>One to many assoication, use <code>getTargetedModificationCollection</code> to get a collection of associated TargetedModification.

</p>
</dd>
</li>
<dt><strong><a href="#genomicsegment">GenomicSegment</a>:</strong>

<dd>
<p>One to many assoication, use <code>getGenomicSegmentCollection</code> to get a collection of associated GenomicSegment.

</p>
</dd>
</li>
<dt><strong><a name="item__22repositoryinfo_22_3a"><a href="#repositoryinfo">RepositoryInfo</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getRepositoryInfo</code> to get the associated RepositoryInfo.

</p>
</dd>
</li>
<dt><strong><a name="item__22histopathology_22_3a"><a href="#histopathology">Histopathology</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getHistopathologyCollection</code> to get a collection of associated Histopathology.

</p>
</dd>
</li>
<dt><strong><a name="item__22cellline_22_3a"><a href="#cellline">CellLine</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getCellLineCollection</code> to get a collection of associated CellLine.

</p>
</dd>
</li>
<dt><strong><a href="#transgene">Transgene</a>:</strong>

<dd>
<p>One to many assoication, use <code>getTransgeneCollection</code> to get a collection of associated Transgene.

</p>
</dd>
</li>
<dt><strong><a name="item__22jaxinfo_22_3a"><a href="#jaxinfo">JaxInfo</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getJaxInfo</code> to get the associated JaxInfo.

</p>
</dd>
</li>
<dt><strong><a href="#inducedmutation">InducedMutation</a>:</strong>

<dd>
<p>One to many assoication, use <code>getInducedMutationCollection</code> to get a collection of associated InducedMutation.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="endpointcode">EndpointCode</a></h1>
<p>CaCORE::CaMOD::EndpointCode - Perl extension for EndpointCode.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::EndpointCode is a Perl object representation of the
caBIO EndpointCode object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A endpoint code captures the parameter for measuring success of anti-drug screen test; mean tumor weightm median survival time, etc.

</p>
<p>
</p>
<h2><a name="attributes_of_endpointcode">Attributes of EndpointCode</a></h2>
<p>The following are all the attributes of the EndpointCode object and their data types:

</p>
<dl>
<dt><strong>code</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>description</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_endpointcode">Associations of EndpointCode</a></h2>
<p>The following are all the objects that are associated with the EndpointCode:

</p>
<pre>
  End Associations and related methods

</pre>
<p>
</p>
<hr />
<h1><a name="screeningresult">ScreeningResult</a></h1>
<p>CaCORE::CaMOD::ScreeningResult - Perl extension for ScreeningResult.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::ScreeningResult is a Perl object representation of the
caBIO ScreeningResult object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>A Screening Result captures the results of the tens of thousands of compounds screened for their ability to inhibit the growth of selected yeast strains altered in DNA damage repair or cell cycle control.

</p>
<p>
</p>
<h2><a name="attributes_of_screeningresult">Attributes of ScreeningResult</a></h2>
<p>The following are all the attributes of the ScreeningResult object and their data types:

</p>
<dl>
<dt><strong><a name="item_diffinh">diffinh</a></strong>

<dd>
<p>data type: <code>float</code>

</p>
</dd>
</li>
<dt><strong><a name="item_aveinh">aveinh</a></strong>

<dd>
<p>data type: <code>float</code>

</p>
</dd>
</li>
<dt><strong><a name="item_stage">stage</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
</li>
<dt><strong><a name="item_inhibitionrate">inhibitionRate</a></strong>

<dd>
<p>data type: <code>float</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_screeningresult">Associations of ScreeningResult</a></h2>
<p>The following are all the objects that are associated with the ScreeningResult:

</p>
<dl>
<dt><strong><a href="#treatmentschedule">TreatmentSchedule</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getTreatmentSchedule</code> to get the associated TreatmentSchedule.

</p>
</dd>
</li>
<dt><strong><a href="#agent">Agent</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getAgent</code> to get the associated Agent.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="invivoresult">InvivoResult</a></h1>
<p>CaCORE::CaMOD::InvivoResult - Perl extension for InvivoResult.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::InvivoResult is a Perl object representation of the
caBIO InvivoResult object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>An Invivo Result captures the results of anti-tumor drug screening.

</p>
<p>
</p>
<h2><a name="attributes_of_invivoresult">Attributes of InvivoResult</a></h2>
<p>The following are all the attributes of the InvivoResult object and their data types:

</p>
<dl>
<dt><strong><a name="item_toxicityevalday">toxicityEvalDay</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_evaluationday">evaluationDay</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong><a name="item_toxicitysurvivors">toxicitySurvivors</a></strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_invivoresult">Associations of InvivoResult</a></h2>
<p>The following are all the objects that are associated with the InvivoResult:

</p>
<dl>
<dt><strong><a href="#treatmentschedule">TreatmentSchedule</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getTreatmentSchedule</code> to get the associated TreatmentSchedule.

</p>
</dd>
</li>
<dt><strong><a href="#agent">Agent</a>:</strong>

<dd>
<p>Many to one assoication, use <code>getAgent</code> to get the associated Agent.

</p>
</dd>
</li>
<dt><strong><a name="item__22endpointcode_22_3a"><a href="#endpointcode">EndpointCode</a>:</a></strong>

<dd>
<p>Many to one assoication, use <code>getEndpointCode</code> to get the associated EndpointCode.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="modificationtype">ModificationType</a></h1>
<p>CaCORE::CaMOD::ModificationType - Perl extension for ModificationType.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::ModificationType is a Perl object representation of the
caBIO ModificationType object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_modificationtype">Attributes of ModificationType</a></h2>
<p>The following are all the attributes of the ModificationType object and their data types:

</p>
<dl>
<dt><strong>name</strong>

<dd>
<p>data type: <code>string</code>

</p>
</dd>
</li>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_modificationtype">Associations of ModificationType</a></h2>
<p>The following are all the objects that are associated with the ModificationType:

</p>
<dl>
<dt><strong><a href="#targetedmodification">TargetedModification</a>:</strong>

<dd>
<p>One to many assoication, use <code>getTargetedModificationCollection</code> to get a collection of associated TargetedModification.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="expressionfeature">ExpressionFeature</a></h1>
<p>CaCORE::CaMOD::ExpressionFeature - Perl extension for ExpressionFeature.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::ExpressionFeature is a Perl object representation of the
caBIO ExpressionFeature object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>The expression feature object describes expression pattern of the engineered gene in the cancer model.

</p>
<p>
</p>
<h2><a name="attributes_of_expressionfeature">Attributes of ExpressionFeature</a></h2>
<p>The following are all the attributes of the ExpressionFeature object and their data types:

</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_expressionfeature">Associations of ExpressionFeature</a></h2>
<p>The following are all the objects that are associated with the ExpressionFeature:

</p>
<dl>
<dt><strong><a name="item__22organontology_22_3a"><a href="#organontology">OrganOntology</a>:</a></strong>

<dd>
<p>One to many assoication, use <code>getOrganOntologyCollection</code> to get a collection of associated OrganOntology.

</p>
</dd>
<dd>
<pre>
  End Associations and related methods

</pre>
</dd>
</li>
</dl>
<p>
</p>
<hr />
<h1><a name="cancermodel">CancerModel</a></h1>
<p>CaCORE::CaMOD::CancerModel - Perl extension for CancerModel.

</p>
<p>
</p>
<h2><a name="abstract">Abstract</a></h2>
<p>The CaCORE::CaMOD::CancerModel is a Perl object representation of the
caBIO CancerModel object.

</p>
<p>
</p>
<h2><a name="description">Description</a></h2>
<p>
</p>
<h2><a name="attributes_of_cancermodel">Attributes of CancerModel</a></h2>
<p>The following are all the attributes of the CancerModel object and their data types:

</p>
<dl>
<dt><strong>id</strong>

<dd>
<p>data type: <code>long</code>

</p>
</dd>
<dd>
<pre>
  End Attributes

</pre>
</dd>
</li>
</dl>
<pre>
  Note: Although you can also use the corresponding setter methods to set the
  attribute values, it is not recommended to do so unless you absolutely have
  to change the object's attributes.

</pre>
<p>
</p>
<h2><a name="associations_of_cancermodel">Associations of CancerModel</a></h2>
<p>The following are all the objects that are associated with the CancerModel:

</p>
<pre>
  End Associations and related methods

</pre>

</body>

</html>
